{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.19.5'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "np.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.1.5'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3.3.4'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import matplotlib\n",
    "matplotlib.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Adjust plot size\n",
    "#options(repr.plot.width=16, repr.plot.height=6)\n",
    "\n",
    "# For some reason, if this is in the same cell as the import command, it doesn't work.\n",
    "plt.rcParams[\"figure.figsize\"] = [18.0,8.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.5.4'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import scipy\n",
    "scipy.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy import stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#from pandarallel import pandarallel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pandarallel.initialize(progress_bar=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "default_max_columns=pd.get_option('display.max_columns')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "default_max_rows=pd.get_option('display.max_rows')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "default_precision=pd.get_option('display.precision')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.precision', 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "default_threshold=np.get_printoptions()['threshold']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.set_printoptions(threshold=10000000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read meta data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = pd.read_csv('20210428-EV/metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rand 2 Batch_order</th>\n",
       "      <th>D-plex Sequencing ID</th>\n",
       "      <th>Lexogen Sequencing ID</th>\n",
       "      <th>Diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1</td>\n",
       "      <td>SFHH005a</td>\n",
       "      <td>SFHH006a</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_2</td>\n",
       "      <td>SFHH005b</td>\n",
       "      <td>SFHH006b</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_3</td>\n",
       "      <td>SFHH005c</td>\n",
       "      <td>SFHH006c</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_4</td>\n",
       "      <td>SFHH005d</td>\n",
       "      <td>SFHH006d</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_5</td>\n",
       "      <td>SFHH005e</td>\n",
       "      <td>SFHH006e</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Rand 2 Batch_order D-plex Sequencing ID Lexogen Sequencing ID  \\\n",
       "0                1_1             SFHH005a              SFHH006a   \n",
       "1                1_2             SFHH005b              SFHH006b   \n",
       "2                1_3             SFHH005c              SFHH006c   \n",
       "3                1_4             SFHH005d              SFHH006d   \n",
       "4                1_5             SFHH005e              SFHH006e   \n",
       "\n",
       "                                        Diagnosis  \n",
       "0                              GBM, IDH1R132H WT,  \n",
       "1  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "2                                GBM, IDH-mutant,  \n",
       "3                                GBM, IDH-mutant,  \n",
       "4                              GBM, IDH1R132H WT,  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta.rename(columns={\"Rand 2 Batch_order\": \"subject\"},inplace=True)\n",
    "meta.rename(columns={\"D-plex Sequencing ID\": \"dplex\"},inplace=True)\n",
    "meta.rename(columns={\"Lexogen Sequencing ID\": \"lexogen\"},inplace=True)\n",
    "meta.rename(columns={\"Diagnosis\": \"diag\"},inplace=True)\n",
    "#meta.set_index('subject',inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>dplex</th>\n",
       "      <th>lexogen</th>\n",
       "      <th>diag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1</td>\n",
       "      <td>SFHH005a</td>\n",
       "      <td>SFHH006a</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_2</td>\n",
       "      <td>SFHH005b</td>\n",
       "      <td>SFHH006b</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_3</td>\n",
       "      <td>SFHH005c</td>\n",
       "      <td>SFHH006c</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_4</td>\n",
       "      <td>SFHH005d</td>\n",
       "      <td>SFHH006d</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_5</td>\n",
       "      <td>SFHH005e</td>\n",
       "      <td>SFHH006e</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1_6</td>\n",
       "      <td>SFHH005f</td>\n",
       "      <td>SFHH006f</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1_7</td>\n",
       "      <td>SFHH005g</td>\n",
       "      <td>SFHH006g</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1_8</td>\n",
       "      <td>SFHH005h</td>\n",
       "      <td>SFHH006h</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1_9</td>\n",
       "      <td>SFHH005i</td>\n",
       "      <td>SFHH006i</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1_10</td>\n",
       "      <td>SFHH005j</td>\n",
       "      <td>SFHH006j</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1_11</td>\n",
       "      <td>SFHH005k</td>\n",
       "      <td>SFHH006k</td>\n",
       "      <td>V01 control (S1),,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2_1</td>\n",
       "      <td>SFHH005l</td>\n",
       "      <td>SFHH006l</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2_2</td>\n",
       "      <td>SFHH005m</td>\n",
       "      <td>SFHH006m</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2_3</td>\n",
       "      <td>SFHH005n</td>\n",
       "      <td>SFHH006n</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2_4</td>\n",
       "      <td>SFHH005o</td>\n",
       "      <td>SFHH006o</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2_5</td>\n",
       "      <td>SFHH005p</td>\n",
       "      <td>SFHH006p</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2_6</td>\n",
       "      <td>SFHH005q</td>\n",
       "      <td>SFHH006q</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2_7</td>\n",
       "      <td>SFHH005r</td>\n",
       "      <td>SFHH006r</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2_8</td>\n",
       "      <td>SFHH005s</td>\n",
       "      <td>SFHH006s</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2_9</td>\n",
       "      <td>SFHH005t</td>\n",
       "      <td>SFHH006t</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2_10</td>\n",
       "      <td>SFHH005u</td>\n",
       "      <td>SFHH006u</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2_11</td>\n",
       "      <td>SFHH005v</td>\n",
       "      <td>SFHH006v</td>\n",
       "      <td>blank1 (C1),,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>3_1</td>\n",
       "      <td>SFHH005w</td>\n",
       "      <td>SFHH006w</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3_2</td>\n",
       "      <td>SFHH005x</td>\n",
       "      <td>SFHH006x</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3_3</td>\n",
       "      <td>SFHH005y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3_4</td>\n",
       "      <td>SFHH005z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3_5</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3_6</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH006ab</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>3_7</td>\n",
       "      <td>SFHH005ac</td>\n",
       "      <td>SFHH006ac</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>3_8</td>\n",
       "      <td>SFHH005ad</td>\n",
       "      <td>SFHH006ad</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>3_9</td>\n",
       "      <td>SFHH005ae</td>\n",
       "      <td>SFHH006ae</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>3_10</td>\n",
       "      <td>SFHH005af</td>\n",
       "      <td>SFHH006af</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>3_11</td>\n",
       "      <td>SFHH005ag</td>\n",
       "      <td>SFHH006ag</td>\n",
       "      <td>V01 control (S1),,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>4_1</td>\n",
       "      <td>SFHH005ah</td>\n",
       "      <td>SFHH006ah</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>4_2</td>\n",
       "      <td>SFHH005ai</td>\n",
       "      <td>SFHH006ai</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>4_3</td>\n",
       "      <td>SFHH005aj</td>\n",
       "      <td>SFHH006aj</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>4_4</td>\n",
       "      <td>SFHH005ak</td>\n",
       "      <td>SFHH006ak</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>4_5</td>\n",
       "      <td>SFHH005al</td>\n",
       "      <td>SFHH006al</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>4_6</td>\n",
       "      <td>SFHH005am</td>\n",
       "      <td>SFHH006am</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>4_7</td>\n",
       "      <td>SFHH005an</td>\n",
       "      <td>SFHH006an</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>4_8</td>\n",
       "      <td>SFHH005ao</td>\n",
       "      <td>SFHH006ao</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>4_9</td>\n",
       "      <td>SFHH005ap</td>\n",
       "      <td>SFHH006ap</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>4_10</td>\n",
       "      <td>SFHH005aq</td>\n",
       "      <td>SFHH006aq</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>4_11</td>\n",
       "      <td>SFHH005ar</td>\n",
       "      <td>SFHH006ar</td>\n",
       "      <td>blank 2 (C1),,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   subject      dplex    lexogen  \\\n",
       "0      1_1   SFHH005a   SFHH006a   \n",
       "1      1_2   SFHH005b   SFHH006b   \n",
       "2      1_3   SFHH005c   SFHH006c   \n",
       "3      1_4   SFHH005d   SFHH006d   \n",
       "4      1_5   SFHH005e   SFHH006e   \n",
       "5      1_6   SFHH005f   SFHH006f   \n",
       "6      1_7   SFHH005g   SFHH006g   \n",
       "7      1_8   SFHH005h   SFHH006h   \n",
       "8      1_9   SFHH005i   SFHH006i   \n",
       "9     1_10   SFHH005j   SFHH006j   \n",
       "10    1_11   SFHH005k   SFHH006k   \n",
       "11     2_1   SFHH005l   SFHH006l   \n",
       "12     2_2   SFHH005m   SFHH006m   \n",
       "13     2_3   SFHH005n   SFHH006n   \n",
       "14     2_4   SFHH005o   SFHH006o   \n",
       "15     2_5   SFHH005p   SFHH006p   \n",
       "16     2_6   SFHH005q   SFHH006q   \n",
       "17     2_7   SFHH005r   SFHH006r   \n",
       "18     2_8   SFHH005s   SFHH006s   \n",
       "19     2_9   SFHH005t   SFHH006t   \n",
       "20    2_10   SFHH005u   SFHH006u   \n",
       "21    2_11   SFHH005v   SFHH006v   \n",
       "22     3_1   SFHH005w   SFHH006w   \n",
       "23     3_2   SFHH005x   SFHH006x   \n",
       "24     3_3   SFHH005y   SFHH006y   \n",
       "25     3_4   SFHH005z   SFHH006z   \n",
       "26     3_5  SFHH005aa  SFHH006aa   \n",
       "27     3_6  SFHH005ab  SFHH006ab   \n",
       "28     3_7  SFHH005ac  SFHH006ac   \n",
       "29     3_8  SFHH005ad  SFHH006ad   \n",
       "30     3_9  SFHH005ae  SFHH006ae   \n",
       "31    3_10  SFHH005af  SFHH006af   \n",
       "32    3_11  SFHH005ag  SFHH006ag   \n",
       "33     4_1  SFHH005ah  SFHH006ah   \n",
       "34     4_2  SFHH005ai  SFHH006ai   \n",
       "35     4_3  SFHH005aj  SFHH006aj   \n",
       "36     4_4  SFHH005ak  SFHH006ak   \n",
       "37     4_5  SFHH005al  SFHH006al   \n",
       "38     4_6  SFHH005am  SFHH006am   \n",
       "39     4_7  SFHH005an  SFHH006an   \n",
       "40     4_8  SFHH005ao  SFHH006ao   \n",
       "41     4_9  SFHH005ap  SFHH006ap   \n",
       "42    4_10  SFHH005aq  SFHH006aq   \n",
       "43    4_11  SFHH005ar  SFHH006ar   \n",
       "\n",
       "                                              diag  \n",
       "0                               GBM, IDH1R132H WT,  \n",
       "1   Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "2                                 GBM, IDH-mutant,  \n",
       "3                                 GBM, IDH-mutant,  \n",
       "4                               GBM, IDH1R132H WT,  \n",
       "5                 Diffuse Astrocytoma, IDH-mutant,  \n",
       "6   Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "7                               GBM, IDH1R132H WT,  \n",
       "8                                 GBM, IDH-mutant,  \n",
       "9                 Diffuse Astrocytoma, IDH-mutant,  \n",
       "10                              V01 control (S1),,  \n",
       "11                Diffuse Astrocytoma, IDH-mutant,  \n",
       "12                              GBM, IDH1R132H WT,  \n",
       "13                Diffuse Astrocytoma, IDH-mutant,  \n",
       "14                              GBM, IDH1R132H WT,  \n",
       "15                Diffuse Astrocytoma, IDH-mutant,  \n",
       "16                                GBM, IDH-mutant,  \n",
       "17                                GBM, IDH-mutant,  \n",
       "18  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "19  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "20                              GBM, IDH1R132H WT,  \n",
       "21                                   blank1 (C1),,  \n",
       "22                                GBM, IDH-mutant,  \n",
       "23  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "24                Diffuse Astrocytoma, IDH-mutant,  \n",
       "25                                GBM, IDH-mutant,  \n",
       "26                                GBM, IDH-mutant,  \n",
       "27                Diffuse Astrocytoma, IDH-mutant,  \n",
       "28  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "29                Diffuse Astrocytoma, IDH-mutant,  \n",
       "30                Diffuse Astrocytoma, IDH-mutant,  \n",
       "31                                GBM, IDH-mutant,  \n",
       "32                              V01 control (S1),,  \n",
       "33  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "34                              GBM, IDH1R132H WT,  \n",
       "35                                GBM, IDH-mutant,  \n",
       "36                              GBM, IDH1R132H WT,  \n",
       "37  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "38                              GBM, IDH1R132H WT,  \n",
       "39                              GBM, IDH1R132H WT,  \n",
       "40  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "41  Oligodendroglioma, IDH-mutant, 1p19q codeleted  \n",
       "42                Diffuse Astrocytoma, IDH-mutant,  \n",
       "43                                  blank 2 (C1),,  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta[['disease','idh','x1p19q']] = meta['diag'].str.split(\",\", 2, expand=True)\n",
    "#meta.drop('diag',axis='columns',inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subject</th>\n",
       "      <th>dplex</th>\n",
       "      <th>lexogen</th>\n",
       "      <th>diag</th>\n",
       "      <th>disease</th>\n",
       "      <th>idh</th>\n",
       "      <th>x1p19q</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1</td>\n",
       "      <td>SFHH005a</td>\n",
       "      <td>SFHH006a</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_2</td>\n",
       "      <td>SFHH005b</td>\n",
       "      <td>SFHH006b</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_3</td>\n",
       "      <td>SFHH005c</td>\n",
       "      <td>SFHH006c</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_4</td>\n",
       "      <td>SFHH005d</td>\n",
       "      <td>SFHH006d</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_5</td>\n",
       "      <td>SFHH005e</td>\n",
       "      <td>SFHH006e</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1_6</td>\n",
       "      <td>SFHH005f</td>\n",
       "      <td>SFHH006f</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1_7</td>\n",
       "      <td>SFHH005g</td>\n",
       "      <td>SFHH006g</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1_8</td>\n",
       "      <td>SFHH005h</td>\n",
       "      <td>SFHH006h</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1_9</td>\n",
       "      <td>SFHH005i</td>\n",
       "      <td>SFHH006i</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1_10</td>\n",
       "      <td>SFHH005j</td>\n",
       "      <td>SFHH006j</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1_11</td>\n",
       "      <td>SFHH005k</td>\n",
       "      <td>SFHH006k</td>\n",
       "      <td>V01 control (S1),,</td>\n",
       "      <td>V01 control (S1)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2_1</td>\n",
       "      <td>SFHH005l</td>\n",
       "      <td>SFHH006l</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2_2</td>\n",
       "      <td>SFHH005m</td>\n",
       "      <td>SFHH006m</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2_3</td>\n",
       "      <td>SFHH005n</td>\n",
       "      <td>SFHH006n</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2_4</td>\n",
       "      <td>SFHH005o</td>\n",
       "      <td>SFHH006o</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2_5</td>\n",
       "      <td>SFHH005p</td>\n",
       "      <td>SFHH006p</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2_6</td>\n",
       "      <td>SFHH005q</td>\n",
       "      <td>SFHH006q</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2_7</td>\n",
       "      <td>SFHH005r</td>\n",
       "      <td>SFHH006r</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2_8</td>\n",
       "      <td>SFHH005s</td>\n",
       "      <td>SFHH006s</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2_9</td>\n",
       "      <td>SFHH005t</td>\n",
       "      <td>SFHH006t</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2_10</td>\n",
       "      <td>SFHH005u</td>\n",
       "      <td>SFHH006u</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2_11</td>\n",
       "      <td>SFHH005v</td>\n",
       "      <td>SFHH006v</td>\n",
       "      <td>blank1 (C1),,</td>\n",
       "      <td>blank1 (C1)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>3_1</td>\n",
       "      <td>SFHH005w</td>\n",
       "      <td>SFHH006w</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>3_2</td>\n",
       "      <td>SFHH005x</td>\n",
       "      <td>SFHH006x</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>3_3</td>\n",
       "      <td>SFHH005y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>3_4</td>\n",
       "      <td>SFHH005z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>3_5</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>3_6</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH006ab</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>3_7</td>\n",
       "      <td>SFHH005ac</td>\n",
       "      <td>SFHH006ac</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>3_8</td>\n",
       "      <td>SFHH005ad</td>\n",
       "      <td>SFHH006ad</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>3_9</td>\n",
       "      <td>SFHH005ae</td>\n",
       "      <td>SFHH006ae</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>3_10</td>\n",
       "      <td>SFHH005af</td>\n",
       "      <td>SFHH006af</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>3_11</td>\n",
       "      <td>SFHH005ag</td>\n",
       "      <td>SFHH006ag</td>\n",
       "      <td>V01 control (S1),,</td>\n",
       "      <td>V01 control (S1)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>4_1</td>\n",
       "      <td>SFHH005ah</td>\n",
       "      <td>SFHH006ah</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>4_2</td>\n",
       "      <td>SFHH005ai</td>\n",
       "      <td>SFHH006ai</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>4_3</td>\n",
       "      <td>SFHH005aj</td>\n",
       "      <td>SFHH006aj</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>4_4</td>\n",
       "      <td>SFHH005ak</td>\n",
       "      <td>SFHH006ak</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>4_5</td>\n",
       "      <td>SFHH005al</td>\n",
       "      <td>SFHH006al</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>4_6</td>\n",
       "      <td>SFHH005am</td>\n",
       "      <td>SFHH006am</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>4_7</td>\n",
       "      <td>SFHH005an</td>\n",
       "      <td>SFHH006an</td>\n",
       "      <td>GBM, IDH1R132H WT,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH1R132H WT</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>4_8</td>\n",
       "      <td>SFHH005ao</td>\n",
       "      <td>SFHH006ao</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>4_9</td>\n",
       "      <td>SFHH005ap</td>\n",
       "      <td>SFHH006ap</td>\n",
       "      <td>Oligodendroglioma, IDH-mutant, 1p19q codeleted</td>\n",
       "      <td>Oligodendroglioma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td>1p19q codeleted</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>4_10</td>\n",
       "      <td>SFHH005aq</td>\n",
       "      <td>SFHH006aq</td>\n",
       "      <td>Diffuse Astrocytoma, IDH-mutant,</td>\n",
       "      <td>Diffuse Astrocytoma</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>4_11</td>\n",
       "      <td>SFHH005ar</td>\n",
       "      <td>SFHH006ar</td>\n",
       "      <td>blank 2 (C1),,</td>\n",
       "      <td>blank 2 (C1)</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   subject      dplex    lexogen  \\\n",
       "0      1_1   SFHH005a   SFHH006a   \n",
       "1      1_2   SFHH005b   SFHH006b   \n",
       "2      1_3   SFHH005c   SFHH006c   \n",
       "3      1_4   SFHH005d   SFHH006d   \n",
       "4      1_5   SFHH005e   SFHH006e   \n",
       "5      1_6   SFHH005f   SFHH006f   \n",
       "6      1_7   SFHH005g   SFHH006g   \n",
       "7      1_8   SFHH005h   SFHH006h   \n",
       "8      1_9   SFHH005i   SFHH006i   \n",
       "9     1_10   SFHH005j   SFHH006j   \n",
       "10    1_11   SFHH005k   SFHH006k   \n",
       "11     2_1   SFHH005l   SFHH006l   \n",
       "12     2_2   SFHH005m   SFHH006m   \n",
       "13     2_3   SFHH005n   SFHH006n   \n",
       "14     2_4   SFHH005o   SFHH006o   \n",
       "15     2_5   SFHH005p   SFHH006p   \n",
       "16     2_6   SFHH005q   SFHH006q   \n",
       "17     2_7   SFHH005r   SFHH006r   \n",
       "18     2_8   SFHH005s   SFHH006s   \n",
       "19     2_9   SFHH005t   SFHH006t   \n",
       "20    2_10   SFHH005u   SFHH006u   \n",
       "21    2_11   SFHH005v   SFHH006v   \n",
       "22     3_1   SFHH005w   SFHH006w   \n",
       "23     3_2   SFHH005x   SFHH006x   \n",
       "24     3_3   SFHH005y   SFHH006y   \n",
       "25     3_4   SFHH005z   SFHH006z   \n",
       "26     3_5  SFHH005aa  SFHH006aa   \n",
       "27     3_6  SFHH005ab  SFHH006ab   \n",
       "28     3_7  SFHH005ac  SFHH006ac   \n",
       "29     3_8  SFHH005ad  SFHH006ad   \n",
       "30     3_9  SFHH005ae  SFHH006ae   \n",
       "31    3_10  SFHH005af  SFHH006af   \n",
       "32    3_11  SFHH005ag  SFHH006ag   \n",
       "33     4_1  SFHH005ah  SFHH006ah   \n",
       "34     4_2  SFHH005ai  SFHH006ai   \n",
       "35     4_3  SFHH005aj  SFHH006aj   \n",
       "36     4_4  SFHH005ak  SFHH006ak   \n",
       "37     4_5  SFHH005al  SFHH006al   \n",
       "38     4_6  SFHH005am  SFHH006am   \n",
       "39     4_7  SFHH005an  SFHH006an   \n",
       "40     4_8  SFHH005ao  SFHH006ao   \n",
       "41     4_9  SFHH005ap  SFHH006ap   \n",
       "42    4_10  SFHH005aq  SFHH006aq   \n",
       "43    4_11  SFHH005ar  SFHH006ar   \n",
       "\n",
       "                                              diag              disease  \\\n",
       "0                               GBM, IDH1R132H WT,                  GBM   \n",
       "1   Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "2                                 GBM, IDH-mutant,                  GBM   \n",
       "3                                 GBM, IDH-mutant,                  GBM   \n",
       "4                               GBM, IDH1R132H WT,                  GBM   \n",
       "5                 Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "6   Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "7                               GBM, IDH1R132H WT,                  GBM   \n",
       "8                                 GBM, IDH-mutant,                  GBM   \n",
       "9                 Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "10                              V01 control (S1),,     V01 control (S1)   \n",
       "11                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "12                              GBM, IDH1R132H WT,                  GBM   \n",
       "13                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "14                              GBM, IDH1R132H WT,                  GBM   \n",
       "15                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "16                                GBM, IDH-mutant,                  GBM   \n",
       "17                                GBM, IDH-mutant,                  GBM   \n",
       "18  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "19  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "20                              GBM, IDH1R132H WT,                  GBM   \n",
       "21                                   blank1 (C1),,          blank1 (C1)   \n",
       "22                                GBM, IDH-mutant,                  GBM   \n",
       "23  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "24                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "25                                GBM, IDH-mutant,                  GBM   \n",
       "26                                GBM, IDH-mutant,                  GBM   \n",
       "27                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "28  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "29                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "30                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "31                                GBM, IDH-mutant,                  GBM   \n",
       "32                              V01 control (S1),,     V01 control (S1)   \n",
       "33  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "34                              GBM, IDH1R132H WT,                  GBM   \n",
       "35                                GBM, IDH-mutant,                  GBM   \n",
       "36                              GBM, IDH1R132H WT,                  GBM   \n",
       "37  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "38                              GBM, IDH1R132H WT,                  GBM   \n",
       "39                              GBM, IDH1R132H WT,                  GBM   \n",
       "40  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "41  Oligodendroglioma, IDH-mutant, 1p19q codeleted    Oligodendroglioma   \n",
       "42                Diffuse Astrocytoma, IDH-mutant,  Diffuse Astrocytoma   \n",
       "43                                  blank 2 (C1),,         blank 2 (C1)   \n",
       "\n",
       "              idh            x1p19q  \n",
       "0    IDH1R132H WT                    \n",
       "1      IDH-mutant   1p19q codeleted  \n",
       "2      IDH-mutant                    \n",
       "3      IDH-mutant                    \n",
       "4    IDH1R132H WT                    \n",
       "5      IDH-mutant                    \n",
       "6      IDH-mutant   1p19q codeleted  \n",
       "7    IDH1R132H WT                    \n",
       "8      IDH-mutant                    \n",
       "9      IDH-mutant                    \n",
       "10                                   \n",
       "11     IDH-mutant                    \n",
       "12   IDH1R132H WT                    \n",
       "13     IDH-mutant                    \n",
       "14   IDH1R132H WT                    \n",
       "15     IDH-mutant                    \n",
       "16     IDH-mutant                    \n",
       "17     IDH-mutant                    \n",
       "18     IDH-mutant   1p19q codeleted  \n",
       "19     IDH-mutant   1p19q codeleted  \n",
       "20   IDH1R132H WT                    \n",
       "21                                   \n",
       "22     IDH-mutant                    \n",
       "23     IDH-mutant   1p19q codeleted  \n",
       "24     IDH-mutant                    \n",
       "25     IDH-mutant                    \n",
       "26     IDH-mutant                    \n",
       "27     IDH-mutant                    \n",
       "28     IDH-mutant   1p19q codeleted  \n",
       "29     IDH-mutant                    \n",
       "30     IDH-mutant                    \n",
       "31     IDH-mutant                    \n",
       "32                                   \n",
       "33     IDH-mutant   1p19q codeleted  \n",
       "34   IDH1R132H WT                    \n",
       "35     IDH-mutant                    \n",
       "36   IDH1R132H WT                    \n",
       "37     IDH-mutant   1p19q codeleted  \n",
       "38   IDH1R132H WT                    \n",
       "39   IDH1R132H WT                    \n",
       "40     IDH-mutant   1p19q codeleted  \n",
       "41     IDH-mutant   1p19q codeleted  \n",
       "42     IDH-mutant                    \n",
       "43                                   "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read raw matrix CORE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df = pd.read_csv('20210428-EV/dplexonlyreport.csv',index_col=0,header=None)\n",
    "df = pd.read_csv('20210428-EV/20210513b-report.csv.gz',index_col=0,header=None,low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(39, 528)"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>...</th>\n",
       "      <th>519</th>\n",
       "      <th>520</th>\n",
       "      <th>521</th>\n",
       "      <th>522</th>\n",
       "      <th>523</th>\n",
       "      <th>524</th>\n",
       "      <th>525</th>\n",
       "      <th>526</th>\n",
       "      <th>527</th>\n",
       "      <th>528</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>...</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>---</th>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Subject</th>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_6</td>\n",
       "      <td>3_6</td>\n",
       "      <td>3_6</td>\n",
       "      <td>3_6</td>\n",
       "      <td>...</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lab kit</th>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>...</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trimmer</th>\n",
       "      <td>bbduk1</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>cutadapt3</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>...</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>cutadapt3</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>cutadapt3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Raw Read Count</th>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>3088240</td>\n",
       "      <td>3088240</td>\n",
       "      <td>3088240</td>\n",
       "      <td>3088240</td>\n",
       "      <td>...</td>\n",
       "      <td>3665258</td>\n",
       "      <td>3665258</td>\n",
       "      <td>3665258</td>\n",
       "      <td>3665258</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Raw Read Length</th>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>...</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trimmed Read Count</th>\n",
       "      <td>1270784</td>\n",
       "      <td>1372973</td>\n",
       "      <td>1372997</td>\n",
       "      <td>1395891</td>\n",
       "      <td>1395893</td>\n",
       "      <td>1395893</td>\n",
       "      <td>1240850</td>\n",
       "      <td>1396651</td>\n",
       "      <td>1396740</td>\n",
       "      <td>1418066</td>\n",
       "      <td>...</td>\n",
       "      <td>3633881</td>\n",
       "      <td>3661993</td>\n",
       "      <td>3664115</td>\n",
       "      <td>3664263</td>\n",
       "      <td>4478092</td>\n",
       "      <td>4823895</td>\n",
       "      <td>4823681</td>\n",
       "      <td>4863961</td>\n",
       "      <td>4867158</td>\n",
       "      <td>4867302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trimmed Ave Read Length</th>\n",
       "      <td>33.8185</td>\n",
       "      <td>38.2306</td>\n",
       "      <td>38.2314</td>\n",
       "      <td>43.5121</td>\n",
       "      <td>43.5142</td>\n",
       "      <td>43.524</td>\n",
       "      <td>40.8779</td>\n",
       "      <td>51.0991</td>\n",
       "      <td>51.101</td>\n",
       "      <td>64.3485</td>\n",
       "      <td>...</td>\n",
       "      <td>32.6867</td>\n",
       "      <td>33.8972</td>\n",
       "      <td>34.0071</td>\n",
       "      <td>34.0507</td>\n",
       "      <td>30.6793</td>\n",
       "      <td>34.7925</td>\n",
       "      <td>34.7983</td>\n",
       "      <td>36.0631</td>\n",
       "      <td>36.1974</td>\n",
       "      <td>36.242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Transcriptome</th>\n",
       "      <td>111104</td>\n",
       "      <td>112499</td>\n",
       "      <td>112492</td>\n",
       "      <td>112689</td>\n",
       "      <td>112696</td>\n",
       "      <td>112689</td>\n",
       "      <td>92673</td>\n",
       "      <td>93752</td>\n",
       "      <td>93727</td>\n",
       "      <td>92285</td>\n",
       "      <td>...</td>\n",
       "      <td>337860</td>\n",
       "      <td>337076</td>\n",
       "      <td>336880</td>\n",
       "      <td>336843</td>\n",
       "      <td>362402</td>\n",
       "      <td>363298</td>\n",
       "      <td>363259</td>\n",
       "      <td>362012</td>\n",
       "      <td>361905</td>\n",
       "      <td>361853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Transcriptome %</th>\n",
       "      <td>8.74</td>\n",
       "      <td>8.19</td>\n",
       "      <td>8.19</td>\n",
       "      <td>8.07</td>\n",
       "      <td>8.07</td>\n",
       "      <td>8.07</td>\n",
       "      <td>7.46</td>\n",
       "      <td>6.71</td>\n",
       "      <td>6.71</td>\n",
       "      <td>6.50</td>\n",
       "      <td>...</td>\n",
       "      <td>9.29</td>\n",
       "      <td>9.20</td>\n",
       "      <td>9.19</td>\n",
       "      <td>9.19</td>\n",
       "      <td>8.09</td>\n",
       "      <td>7.53</td>\n",
       "      <td>7.53</td>\n",
       "      <td>7.44</td>\n",
       "      <td>7.43</td>\n",
       "      <td>7.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Genome</th>\n",
       "      <td>655927</td>\n",
       "      <td>697141</td>\n",
       "      <td>697145</td>\n",
       "      <td>701944</td>\n",
       "      <td>701935</td>\n",
       "      <td>701914</td>\n",
       "      <td>727759</td>\n",
       "      <td>834086</td>\n",
       "      <td>834097</td>\n",
       "      <td>837173</td>\n",
       "      <td>...</td>\n",
       "      <td>1555401</td>\n",
       "      <td>1547893</td>\n",
       "      <td>1547026</td>\n",
       "      <td>1546840</td>\n",
       "      <td>1947074</td>\n",
       "      <td>1730325</td>\n",
       "      <td>1729961</td>\n",
       "      <td>1718029</td>\n",
       "      <td>1716972</td>\n",
       "      <td>1716740</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows  528 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       1          2          3          4    \\\n",
       "0                                                                             \n",
       "                                 SFHH005aa  SFHH005aa  SFHH005aa  SFHH005aa   \n",
       "---                                    ---        ---        ---        ---   \n",
       "Subject                                3_5        3_5        3_5        3_5   \n",
       "Lab kit                             D-plex     D-plex     D-plex     D-plex   \n",
       "Trimmer                             bbduk1     bbduk2     bbduk3  cutadapt1   \n",
       "Raw Read Count                     4038879    4038879    4038879    4038879   \n",
       "Raw Read Length                    35.0646    35.0646    35.0646    35.0646   \n",
       "Trimmed Read Count                 1270784    1372973    1372997    1395891   \n",
       "Trimmed Ave Read Length            33.8185    38.2306    38.2314    43.5121   \n",
       "STAR Aligned to Transcriptome       111104     112499     112492     112689   \n",
       "STAR Aligned to Transcriptome %       8.74       8.19       8.19       8.07   \n",
       "STAR Aligned to Genome              655927     697141     697145     701944   \n",
       "\n",
       "                                       5          6          7          8    \\\n",
       "0                                                                             \n",
       "                                 SFHH005aa  SFHH005aa  SFHH005ab  SFHH005ab   \n",
       "---                                    ---        ---        ---        ---   \n",
       "Subject                                3_5        3_5        3_6        3_6   \n",
       "Lab kit                             D-plex     D-plex     D-plex     D-plex   \n",
       "Trimmer                          cutadapt2  cutadapt3     bbduk1     bbduk2   \n",
       "Raw Read Count                     4038879    4038879    3088240    3088240   \n",
       "Raw Read Length                    35.0646    35.0646    48.6946    48.6946   \n",
       "Trimmed Read Count                 1395893    1395893    1240850    1396651   \n",
       "Trimmed Ave Read Length            43.5142     43.524    40.8779    51.0991   \n",
       "STAR Aligned to Transcriptome       112696     112689      92673      93752   \n",
       "STAR Aligned to Transcriptome %       8.07       8.07       7.46       6.71   \n",
       "STAR Aligned to Genome              701935     701914     727759     834086   \n",
       "\n",
       "                                       9          10   ...       519  \\\n",
       "0                                                      ...             \n",
       "                                 SFHH005ab  SFHH005ab  ...  SFHH006y   \n",
       "---                                    ---        ---  ...       ---   \n",
       "Subject                                3_6        3_6  ...       3_3   \n",
       "Lab kit                             D-plex     D-plex  ...   Lexogen   \n",
       "Trimmer                             bbduk3  cutadapt1  ...    bbduk3   \n",
       "Raw Read Count                     3088240    3088240  ...   3665258   \n",
       "Raw Read Length                    48.6946    48.6946  ...   34.0495   \n",
       "Trimmed Read Count                 1396740    1418066  ...   3633881   \n",
       "Trimmed Ave Read Length             51.101    64.3485  ...   32.6867   \n",
       "STAR Aligned to Transcriptome        93727      92285  ...    337860   \n",
       "STAR Aligned to Transcriptome %       6.71       6.50  ...      9.29   \n",
       "STAR Aligned to Genome              834097     837173  ...   1555401   \n",
       "\n",
       "                                       520        521        522       523  \\\n",
       "0                                                                            \n",
       "                                  SFHH006y   SFHH006y   SFHH006y  SFHH006z   \n",
       "---                                    ---        ---        ---       ---   \n",
       "Subject                                3_3        3_3        3_3       3_4   \n",
       "Lab kit                            Lexogen    Lexogen    Lexogen   Lexogen   \n",
       "Trimmer                          cutadapt1  cutadapt2  cutadapt3    bbduk1   \n",
       "Raw Read Count                     3665258    3665258    3665258   4868169   \n",
       "Raw Read Length                    34.0495    34.0495    34.0495   36.2413   \n",
       "Trimmed Read Count                 3661993    3664115    3664263   4478092   \n",
       "Trimmed Ave Read Length            33.8972    34.0071    34.0507   30.6793   \n",
       "STAR Aligned to Transcriptome       337076     336880     336843    362402   \n",
       "STAR Aligned to Transcriptome %       9.20       9.19       9.19      8.09   \n",
       "STAR Aligned to Genome             1547893    1547026    1546840   1947074   \n",
       "\n",
       "                                      524       525        526        527  \\\n",
       "0                                                                           \n",
       "                                 SFHH006z  SFHH006z   SFHH006z   SFHH006z   \n",
       "---                                   ---       ---        ---        ---   \n",
       "Subject                               3_4       3_4        3_4        3_4   \n",
       "Lab kit                           Lexogen   Lexogen    Lexogen    Lexogen   \n",
       "Trimmer                            bbduk2    bbduk3  cutadapt1  cutadapt2   \n",
       "Raw Read Count                    4868169   4868169    4868169    4868169   \n",
       "Raw Read Length                   36.2413   36.2413    36.2413    36.2413   \n",
       "Trimmed Read Count                4823895   4823681    4863961    4867158   \n",
       "Trimmed Ave Read Length           34.7925   34.7983    36.0631    36.1974   \n",
       "STAR Aligned to Transcriptome      363298    363259     362012     361905   \n",
       "STAR Aligned to Transcriptome %      7.53      7.53       7.44       7.43   \n",
       "STAR Aligned to Genome            1730325   1729961    1718029    1716972   \n",
       "\n",
       "                                       528  \n",
       "0                                           \n",
       "                                  SFHH006z  \n",
       "---                                    ---  \n",
       "Subject                                3_4  \n",
       "Lab kit                            Lexogen  \n",
       "Trimmer                          cutadapt3  \n",
       "Raw Read Count                     4868169  \n",
       "Raw Read Length                    36.2413  \n",
       "Trimmed Read Count                 4867302  \n",
       "Trimmed Ave Read Length             36.242  \n",
       "STAR Aligned to Transcriptome       361853  \n",
       "STAR Aligned to Transcriptome %       7.43  \n",
       "STAR Aligned to Genome             1716740  \n",
       "\n",
       "[12 rows x 528 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop(\"---\",inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(38, 528)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.rename({'  ':'sample'},axis='rows',inplace=True)\n",
    "df.rename({'Subject':'subject'},axis='rows',inplace=True)\n",
    "df.rename({'Trimmer':'trimmer'},axis='rows',inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>...</th>\n",
       "      <th>519</th>\n",
       "      <th>520</th>\n",
       "      <th>521</th>\n",
       "      <th>522</th>\n",
       "      <th>523</th>\n",
       "      <th>524</th>\n",
       "      <th>525</th>\n",
       "      <th>526</th>\n",
       "      <th>527</th>\n",
       "      <th>528</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>SFHH005ab</td>\n",
       "      <td>...</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006y</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "      <td>SFHH006z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subject</th>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_5</td>\n",
       "      <td>3_6</td>\n",
       "      <td>3_6</td>\n",
       "      <td>3_6</td>\n",
       "      <td>3_6</td>\n",
       "      <td>...</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_3</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "      <td>3_4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lab kit</th>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>...</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "      <td>Lexogen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmer</th>\n",
       "      <td>bbduk1</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>cutadapt3</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>...</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>cutadapt3</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>cutadapt3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Raw Read Count</th>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>4038879</td>\n",
       "      <td>3088240</td>\n",
       "      <td>3088240</td>\n",
       "      <td>3088240</td>\n",
       "      <td>3088240</td>\n",
       "      <td>...</td>\n",
       "      <td>3665258</td>\n",
       "      <td>3665258</td>\n",
       "      <td>3665258</td>\n",
       "      <td>3665258</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "      <td>4868169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Raw Read Length</th>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>48.6946</td>\n",
       "      <td>...</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>34.0495</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "      <td>36.2413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trimmed Read Count</th>\n",
       "      <td>1270784</td>\n",
       "      <td>1372973</td>\n",
       "      <td>1372997</td>\n",
       "      <td>1395891</td>\n",
       "      <td>1395893</td>\n",
       "      <td>1395893</td>\n",
       "      <td>1240850</td>\n",
       "      <td>1396651</td>\n",
       "      <td>1396740</td>\n",
       "      <td>1418066</td>\n",
       "      <td>...</td>\n",
       "      <td>3633881</td>\n",
       "      <td>3661993</td>\n",
       "      <td>3664115</td>\n",
       "      <td>3664263</td>\n",
       "      <td>4478092</td>\n",
       "      <td>4823895</td>\n",
       "      <td>4823681</td>\n",
       "      <td>4863961</td>\n",
       "      <td>4867158</td>\n",
       "      <td>4867302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trimmed Ave Read Length</th>\n",
       "      <td>33.8185</td>\n",
       "      <td>38.2306</td>\n",
       "      <td>38.2314</td>\n",
       "      <td>43.5121</td>\n",
       "      <td>43.5142</td>\n",
       "      <td>43.524</td>\n",
       "      <td>40.8779</td>\n",
       "      <td>51.0991</td>\n",
       "      <td>51.101</td>\n",
       "      <td>64.3485</td>\n",
       "      <td>...</td>\n",
       "      <td>32.6867</td>\n",
       "      <td>33.8972</td>\n",
       "      <td>34.0071</td>\n",
       "      <td>34.0507</td>\n",
       "      <td>30.6793</td>\n",
       "      <td>34.7925</td>\n",
       "      <td>34.7983</td>\n",
       "      <td>36.0631</td>\n",
       "      <td>36.1974</td>\n",
       "      <td>36.242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Transcriptome</th>\n",
       "      <td>111104</td>\n",
       "      <td>112499</td>\n",
       "      <td>112492</td>\n",
       "      <td>112689</td>\n",
       "      <td>112696</td>\n",
       "      <td>112689</td>\n",
       "      <td>92673</td>\n",
       "      <td>93752</td>\n",
       "      <td>93727</td>\n",
       "      <td>92285</td>\n",
       "      <td>...</td>\n",
       "      <td>337860</td>\n",
       "      <td>337076</td>\n",
       "      <td>336880</td>\n",
       "      <td>336843</td>\n",
       "      <td>362402</td>\n",
       "      <td>363298</td>\n",
       "      <td>363259</td>\n",
       "      <td>362012</td>\n",
       "      <td>361905</td>\n",
       "      <td>361853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Transcriptome %</th>\n",
       "      <td>8.74</td>\n",
       "      <td>8.19</td>\n",
       "      <td>8.19</td>\n",
       "      <td>8.07</td>\n",
       "      <td>8.07</td>\n",
       "      <td>8.07</td>\n",
       "      <td>7.46</td>\n",
       "      <td>6.71</td>\n",
       "      <td>6.71</td>\n",
       "      <td>6.50</td>\n",
       "      <td>...</td>\n",
       "      <td>9.29</td>\n",
       "      <td>9.20</td>\n",
       "      <td>9.19</td>\n",
       "      <td>9.19</td>\n",
       "      <td>8.09</td>\n",
       "      <td>7.53</td>\n",
       "      <td>7.53</td>\n",
       "      <td>7.44</td>\n",
       "      <td>7.43</td>\n",
       "      <td>7.43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Genome</th>\n",
       "      <td>655927</td>\n",
       "      <td>697141</td>\n",
       "      <td>697145</td>\n",
       "      <td>701944</td>\n",
       "      <td>701935</td>\n",
       "      <td>701914</td>\n",
       "      <td>727759</td>\n",
       "      <td>834086</td>\n",
       "      <td>834097</td>\n",
       "      <td>837173</td>\n",
       "      <td>...</td>\n",
       "      <td>1555401</td>\n",
       "      <td>1547893</td>\n",
       "      <td>1547026</td>\n",
       "      <td>1546840</td>\n",
       "      <td>1947074</td>\n",
       "      <td>1730325</td>\n",
       "      <td>1729961</td>\n",
       "      <td>1718029</td>\n",
       "      <td>1716972</td>\n",
       "      <td>1716740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to Genome %</th>\n",
       "      <td>51.61</td>\n",
       "      <td>50.77</td>\n",
       "      <td>50.77</td>\n",
       "      <td>50.28</td>\n",
       "      <td>50.28</td>\n",
       "      <td>50.28</td>\n",
       "      <td>58.65</td>\n",
       "      <td>59.72</td>\n",
       "      <td>59.71</td>\n",
       "      <td>59.03</td>\n",
       "      <td>...</td>\n",
       "      <td>42.80</td>\n",
       "      <td>42.26</td>\n",
       "      <td>42.22</td>\n",
       "      <td>42.21</td>\n",
       "      <td>43.47</td>\n",
       "      <td>35.86</td>\n",
       "      <td>35.86</td>\n",
       "      <td>35.32</td>\n",
       "      <td>35.27</td>\n",
       "      <td>35.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Unaligned</th>\n",
       "      <td>614857</td>\n",
       "      <td>675832</td>\n",
       "      <td>675852</td>\n",
       "      <td>693947</td>\n",
       "      <td>693958</td>\n",
       "      <td>693979</td>\n",
       "      <td>513091</td>\n",
       "      <td>562565</td>\n",
       "      <td>562643</td>\n",
       "      <td>580893</td>\n",
       "      <td>...</td>\n",
       "      <td>2078480</td>\n",
       "      <td>2114100</td>\n",
       "      <td>2117089</td>\n",
       "      <td>2117423</td>\n",
       "      <td>2531018</td>\n",
       "      <td>3093570</td>\n",
       "      <td>3093720</td>\n",
       "      <td>3145932</td>\n",
       "      <td>3150186</td>\n",
       "      <td>3150562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Unaligned %</th>\n",
       "      <td>48.38</td>\n",
       "      <td>49.22</td>\n",
       "      <td>49.22</td>\n",
       "      <td>49.71</td>\n",
       "      <td>49.71</td>\n",
       "      <td>49.71</td>\n",
       "      <td>41.34</td>\n",
       "      <td>40.27</td>\n",
       "      <td>40.28</td>\n",
       "      <td>40.96</td>\n",
       "      <td>...</td>\n",
       "      <td>57.19</td>\n",
       "      <td>57.73</td>\n",
       "      <td>57.77</td>\n",
       "      <td>57.78</td>\n",
       "      <td>56.52</td>\n",
       "      <td>64.13</td>\n",
       "      <td>64.13</td>\n",
       "      <td>64.67</td>\n",
       "      <td>64.72</td>\n",
       "      <td>64.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Unmapped</th>\n",
       "      <td>614857</td>\n",
       "      <td>675832</td>\n",
       "      <td>675852</td>\n",
       "      <td>693947</td>\n",
       "      <td>693958</td>\n",
       "      <td>693979</td>\n",
       "      <td>513091</td>\n",
       "      <td>562565</td>\n",
       "      <td>562643</td>\n",
       "      <td>580893</td>\n",
       "      <td>...</td>\n",
       "      <td>2078480</td>\n",
       "      <td>2114100</td>\n",
       "      <td>2117089</td>\n",
       "      <td>2117423</td>\n",
       "      <td>2531018</td>\n",
       "      <td>3093570</td>\n",
       "      <td>3093720</td>\n",
       "      <td>3145932</td>\n",
       "      <td>3150186</td>\n",
       "      <td>3150562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Unmapped %</th>\n",
       "      <td>48.38</td>\n",
       "      <td>49.22</td>\n",
       "      <td>49.22</td>\n",
       "      <td>49.71</td>\n",
       "      <td>49.71</td>\n",
       "      <td>49.71</td>\n",
       "      <td>41.34</td>\n",
       "      <td>40.27</td>\n",
       "      <td>40.28</td>\n",
       "      <td>40.96</td>\n",
       "      <td>...</td>\n",
       "      <td>57.19</td>\n",
       "      <td>57.73</td>\n",
       "      <td>57.77</td>\n",
       "      <td>57.78</td>\n",
       "      <td>56.52</td>\n",
       "      <td>64.13</td>\n",
       "      <td>64.13</td>\n",
       "      <td>64.67</td>\n",
       "      <td>64.72</td>\n",
       "      <td>64.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to hg38 (1)</th>\n",
       "      <td>165418</td>\n",
       "      <td>204662</td>\n",
       "      <td>204672</td>\n",
       "      <td>208906</td>\n",
       "      <td>208905</td>\n",
       "      <td>208947</td>\n",
       "      <td>475443</td>\n",
       "      <td>600590</td>\n",
       "      <td>600639</td>\n",
       "      <td>609164</td>\n",
       "      <td>...</td>\n",
       "      <td>46809</td>\n",
       "      <td>57182</td>\n",
       "      <td>58275</td>\n",
       "      <td>59209</td>\n",
       "      <td>40727</td>\n",
       "      <td>47736</td>\n",
       "      <td>47746</td>\n",
       "      <td>62246</td>\n",
       "      <td>63840</td>\n",
       "      <td>65105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to hg38 (1) %</th>\n",
       "      <td>13.01</td>\n",
       "      <td>14.90</td>\n",
       "      <td>14.90</td>\n",
       "      <td>14.96</td>\n",
       "      <td>14.96</td>\n",
       "      <td>14.96</td>\n",
       "      <td>38.31</td>\n",
       "      <td>43.00</td>\n",
       "      <td>43.00</td>\n",
       "      <td>42.95</td>\n",
       "      <td>...</td>\n",
       "      <td>1.28</td>\n",
       "      <td>1.56</td>\n",
       "      <td>1.59</td>\n",
       "      <td>1.61</td>\n",
       "      <td>.90</td>\n",
       "      <td>.98</td>\n",
       "      <td>.98</td>\n",
       "      <td>1.27</td>\n",
       "      <td>1.31</td>\n",
       "      <td>1.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to mirna</th>\n",
       "      <td>156554</td>\n",
       "      <td>154884</td>\n",
       "      <td>154884</td>\n",
       "      <td>156555</td>\n",
       "      <td>156554</td>\n",
       "      <td>156554</td>\n",
       "      <td>125155</td>\n",
       "      <td>115596</td>\n",
       "      <td>115600</td>\n",
       "      <td>114309</td>\n",
       "      <td>...</td>\n",
       "      <td>328284</td>\n",
       "      <td>326524</td>\n",
       "      <td>326446</td>\n",
       "      <td>326417</td>\n",
       "      <td>358302</td>\n",
       "      <td>382995</td>\n",
       "      <td>382987</td>\n",
       "      <td>380420</td>\n",
       "      <td>380500</td>\n",
       "      <td>380475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAR Aligned to mirna %</th>\n",
       "      <td>12.31</td>\n",
       "      <td>11.28</td>\n",
       "      <td>11.28</td>\n",
       "      <td>11.21</td>\n",
       "      <td>11.21</td>\n",
       "      <td>11.21</td>\n",
       "      <td>10.08</td>\n",
       "      <td>8.27</td>\n",
       "      <td>8.27</td>\n",
       "      <td>8.06</td>\n",
       "      <td>...</td>\n",
       "      <td>9.03</td>\n",
       "      <td>8.91</td>\n",
       "      <td>8.90</td>\n",
       "      <td>8.90</td>\n",
       "      <td>8.00</td>\n",
       "      <td>7.93</td>\n",
       "      <td>7.93</td>\n",
       "      <td>7.82</td>\n",
       "      <td>7.81</td>\n",
       "      <td>7.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie Aligned to mirna</th>\n",
       "      <td>49775</td>\n",
       "      <td>49507</td>\n",
       "      <td>49507</td>\n",
       "      <td>50782</td>\n",
       "      <td>50781</td>\n",
       "      <td>50781</td>\n",
       "      <td>39999</td>\n",
       "      <td>36092</td>\n",
       "      <td>36092</td>\n",
       "      <td>36284</td>\n",
       "      <td>...</td>\n",
       "      <td>68521</td>\n",
       "      <td>67732</td>\n",
       "      <td>67692</td>\n",
       "      <td>67685</td>\n",
       "      <td>68521</td>\n",
       "      <td>66796</td>\n",
       "      <td>66795</td>\n",
       "      <td>65868</td>\n",
       "      <td>65889</td>\n",
       "      <td>65886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie Aligned to mirna %</th>\n",
       "      <td>3.91</td>\n",
       "      <td>3.60</td>\n",
       "      <td>3.60</td>\n",
       "      <td>3.63</td>\n",
       "      <td>3.63</td>\n",
       "      <td>3.63</td>\n",
       "      <td>3.22</td>\n",
       "      <td>2.58</td>\n",
       "      <td>2.58</td>\n",
       "      <td>2.55</td>\n",
       "      <td>...</td>\n",
       "      <td>1.88</td>\n",
       "      <td>1.84</td>\n",
       "      <td>1.84</td>\n",
       "      <td>1.84</td>\n",
       "      <td>1.53</td>\n",
       "      <td>1.38</td>\n",
       "      <td>1.38</td>\n",
       "      <td>1.35</td>\n",
       "      <td>1.35</td>\n",
       "      <td>1.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to mirna</th>\n",
       "      <td>946</td>\n",
       "      <td>1060</td>\n",
       "      <td>1060</td>\n",
       "      <td>1235</td>\n",
       "      <td>1235</td>\n",
       "      <td>1235</td>\n",
       "      <td>4737</td>\n",
       "      <td>5302</td>\n",
       "      <td>5302</td>\n",
       "      <td>5708</td>\n",
       "      <td>...</td>\n",
       "      <td>15195</td>\n",
       "      <td>15288</td>\n",
       "      <td>15268</td>\n",
       "      <td>15268</td>\n",
       "      <td>14483</td>\n",
       "      <td>15281</td>\n",
       "      <td>15282</td>\n",
       "      <td>15345</td>\n",
       "      <td>15333</td>\n",
       "      <td>15333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to mirna %</th>\n",
       "      <td>.07</td>\n",
       "      <td>.07</td>\n",
       "      <td>.07</td>\n",
       "      <td>.08</td>\n",
       "      <td>.08</td>\n",
       "      <td>.08</td>\n",
       "      <td>.38</td>\n",
       "      <td>.37</td>\n",
       "      <td>.37</td>\n",
       "      <td>.40</td>\n",
       "      <td>...</td>\n",
       "      <td>.41</td>\n",
       "      <td>.41</td>\n",
       "      <td>.41</td>\n",
       "      <td>.41</td>\n",
       "      <td>.32</td>\n",
       "      <td>.31</td>\n",
       "      <td>.31</td>\n",
       "      <td>.31</td>\n",
       "      <td>.31</td>\n",
       "      <td>.31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to RMSK</th>\n",
       "      <td>92670</td>\n",
       "      <td>116878</td>\n",
       "      <td>116879</td>\n",
       "      <td>121972</td>\n",
       "      <td>121971</td>\n",
       "      <td>121999</td>\n",
       "      <td>304510</td>\n",
       "      <td>388795</td>\n",
       "      <td>388827</td>\n",
       "      <td>402988</td>\n",
       "      <td>...</td>\n",
       "      <td>24856</td>\n",
       "      <td>31291</td>\n",
       "      <td>31914</td>\n",
       "      <td>32367</td>\n",
       "      <td>18268</td>\n",
       "      <td>23241</td>\n",
       "      <td>23246</td>\n",
       "      <td>32052</td>\n",
       "      <td>32955</td>\n",
       "      <td>33561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to RMSK %</th>\n",
       "      <td>7.29</td>\n",
       "      <td>8.51</td>\n",
       "      <td>8.51</td>\n",
       "      <td>8.73</td>\n",
       "      <td>8.73</td>\n",
       "      <td>8.73</td>\n",
       "      <td>24.54</td>\n",
       "      <td>27.83</td>\n",
       "      <td>27.83</td>\n",
       "      <td>28.41</td>\n",
       "      <td>...</td>\n",
       "      <td>.68</td>\n",
       "      <td>.85</td>\n",
       "      <td>.87</td>\n",
       "      <td>.88</td>\n",
       "      <td>.40</td>\n",
       "      <td>.48</td>\n",
       "      <td>.48</td>\n",
       "      <td>.65</td>\n",
       "      <td>.67</td>\n",
       "      <td>.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to phiX</th>\n",
       "      <td>7745</td>\n",
       "      <td>9144</td>\n",
       "      <td>9144</td>\n",
       "      <td>11745</td>\n",
       "      <td>11747</td>\n",
       "      <td>11745</td>\n",
       "      <td>40687</td>\n",
       "      <td>47810</td>\n",
       "      <td>47811</td>\n",
       "      <td>55340</td>\n",
       "      <td>...</td>\n",
       "      <td>1303</td>\n",
       "      <td>10874</td>\n",
       "      <td>10896</td>\n",
       "      <td>10898</td>\n",
       "      <td>1495</td>\n",
       "      <td>1878</td>\n",
       "      <td>1878</td>\n",
       "      <td>14144</td>\n",
       "      <td>14180</td>\n",
       "      <td>14182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to phiX %</th>\n",
       "      <td>.60</td>\n",
       "      <td>.66</td>\n",
       "      <td>.66</td>\n",
       "      <td>.84</td>\n",
       "      <td>.84</td>\n",
       "      <td>.84</td>\n",
       "      <td>3.27</td>\n",
       "      <td>3.42</td>\n",
       "      <td>3.42</td>\n",
       "      <td>3.90</td>\n",
       "      <td>...</td>\n",
       "      <td>.03</td>\n",
       "      <td>.29</td>\n",
       "      <td>.29</td>\n",
       "      <td>.29</td>\n",
       "      <td>.03</td>\n",
       "      <td>.03</td>\n",
       "      <td>.03</td>\n",
       "      <td>.29</td>\n",
       "      <td>.29</td>\n",
       "      <td>.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to Salmonella</th>\n",
       "      <td>45946</td>\n",
       "      <td>60133</td>\n",
       "      <td>60133</td>\n",
       "      <td>61284</td>\n",
       "      <td>61285</td>\n",
       "      <td>61282</td>\n",
       "      <td>43007</td>\n",
       "      <td>54722</td>\n",
       "      <td>54722</td>\n",
       "      <td>56558</td>\n",
       "      <td>...</td>\n",
       "      <td>69276</td>\n",
       "      <td>70037</td>\n",
       "      <td>70056</td>\n",
       "      <td>70047</td>\n",
       "      <td>76595</td>\n",
       "      <td>116173</td>\n",
       "      <td>116172</td>\n",
       "      <td>117387</td>\n",
       "      <td>117383</td>\n",
       "      <td>117374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to Salmonella %</th>\n",
       "      <td>3.61</td>\n",
       "      <td>4.37</td>\n",
       "      <td>4.37</td>\n",
       "      <td>4.39</td>\n",
       "      <td>4.39</td>\n",
       "      <td>4.39</td>\n",
       "      <td>3.46</td>\n",
       "      <td>3.91</td>\n",
       "      <td>3.91</td>\n",
       "      <td>3.98</td>\n",
       "      <td>...</td>\n",
       "      <td>1.90</td>\n",
       "      <td>1.91</td>\n",
       "      <td>1.91</td>\n",
       "      <td>1.91</td>\n",
       "      <td>1.71</td>\n",
       "      <td>2.40</td>\n",
       "      <td>2.40</td>\n",
       "      <td>2.41</td>\n",
       "      <td>2.41</td>\n",
       "      <td>2.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to masked Salmonella</th>\n",
       "      <td>14700</td>\n",
       "      <td>17811</td>\n",
       "      <td>17811</td>\n",
       "      <td>18218</td>\n",
       "      <td>18218</td>\n",
       "      <td>18215</td>\n",
       "      <td>14998</td>\n",
       "      <td>17504</td>\n",
       "      <td>17504</td>\n",
       "      <td>18046</td>\n",
       "      <td>...</td>\n",
       "      <td>34220</td>\n",
       "      <td>34653</td>\n",
       "      <td>34689</td>\n",
       "      <td>34684</td>\n",
       "      <td>31307</td>\n",
       "      <td>58894</td>\n",
       "      <td>58894</td>\n",
       "      <td>59592</td>\n",
       "      <td>59593</td>\n",
       "      <td>59589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to masked Salmonella %</th>\n",
       "      <td>1.15</td>\n",
       "      <td>1.29</td>\n",
       "      <td>1.29</td>\n",
       "      <td>1.30</td>\n",
       "      <td>1.30</td>\n",
       "      <td>1.30</td>\n",
       "      <td>1.20</td>\n",
       "      <td>1.25</td>\n",
       "      <td>1.25</td>\n",
       "      <td>1.27</td>\n",
       "      <td>...</td>\n",
       "      <td>.94</td>\n",
       "      <td>.94</td>\n",
       "      <td>.94</td>\n",
       "      <td>.94</td>\n",
       "      <td>.69</td>\n",
       "      <td>1.22</td>\n",
       "      <td>1.22</td>\n",
       "      <td>1.22</td>\n",
       "      <td>1.22</td>\n",
       "      <td>1.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to Burkholderia</th>\n",
       "      <td>375543</td>\n",
       "      <td>413414</td>\n",
       "      <td>413414</td>\n",
       "      <td>420290</td>\n",
       "      <td>420299</td>\n",
       "      <td>420297</td>\n",
       "      <td>274240</td>\n",
       "      <td>303638</td>\n",
       "      <td>303638</td>\n",
       "      <td>308657</td>\n",
       "      <td>...</td>\n",
       "      <td>2101944</td>\n",
       "      <td>2121875</td>\n",
       "      <td>2118401</td>\n",
       "      <td>2118297</td>\n",
       "      <td>2622229</td>\n",
       "      <td>2985241</td>\n",
       "      <td>2984974</td>\n",
       "      <td>3016302</td>\n",
       "      <td>3009621</td>\n",
       "      <td>3009486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to Burkholderia %</th>\n",
       "      <td>29.55</td>\n",
       "      <td>30.11</td>\n",
       "      <td>30.11</td>\n",
       "      <td>30.10</td>\n",
       "      <td>30.10</td>\n",
       "      <td>30.10</td>\n",
       "      <td>22.10</td>\n",
       "      <td>21.74</td>\n",
       "      <td>21.73</td>\n",
       "      <td>21.76</td>\n",
       "      <td>...</td>\n",
       "      <td>57.84</td>\n",
       "      <td>57.94</td>\n",
       "      <td>57.81</td>\n",
       "      <td>57.80</td>\n",
       "      <td>58.55</td>\n",
       "      <td>61.88</td>\n",
       "      <td>61.88</td>\n",
       "      <td>62.01</td>\n",
       "      <td>61.83</td>\n",
       "      <td>61.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to masked Burkholderia</th>\n",
       "      <td>277406</td>\n",
       "      <td>298416</td>\n",
       "      <td>298416</td>\n",
       "      <td>304330</td>\n",
       "      <td>304339</td>\n",
       "      <td>304338</td>\n",
       "      <td>180637</td>\n",
       "      <td>197994</td>\n",
       "      <td>197994</td>\n",
       "      <td>201850</td>\n",
       "      <td>...</td>\n",
       "      <td>1961758</td>\n",
       "      <td>1980757</td>\n",
       "      <td>1977308</td>\n",
       "      <td>1977214</td>\n",
       "      <td>2433977</td>\n",
       "      <td>2777831</td>\n",
       "      <td>2777568</td>\n",
       "      <td>2807588</td>\n",
       "      <td>2800958</td>\n",
       "      <td>2800836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to masked Burkholderia %</th>\n",
       "      <td>21.82</td>\n",
       "      <td>21.73</td>\n",
       "      <td>21.73</td>\n",
       "      <td>21.80</td>\n",
       "      <td>21.80</td>\n",
       "      <td>21.80</td>\n",
       "      <td>14.55</td>\n",
       "      <td>14.17</td>\n",
       "      <td>14.17</td>\n",
       "      <td>14.23</td>\n",
       "      <td>...</td>\n",
       "      <td>53.98</td>\n",
       "      <td>54.08</td>\n",
       "      <td>53.96</td>\n",
       "      <td>53.95</td>\n",
       "      <td>54.35</td>\n",
       "      <td>57.58</td>\n",
       "      <td>57.58</td>\n",
       "      <td>57.72</td>\n",
       "      <td>57.54</td>\n",
       "      <td>57.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to mRNA_Prot</th>\n",
       "      <td>58736</td>\n",
       "      <td>71735</td>\n",
       "      <td>71735</td>\n",
       "      <td>75518</td>\n",
       "      <td>75522</td>\n",
       "      <td>75517</td>\n",
       "      <td>187948</td>\n",
       "      <td>230807</td>\n",
       "      <td>230826</td>\n",
       "      <td>241994</td>\n",
       "      <td>...</td>\n",
       "      <td>38124</td>\n",
       "      <td>38784</td>\n",
       "      <td>38798</td>\n",
       "      <td>38831</td>\n",
       "      <td>27716</td>\n",
       "      <td>32278</td>\n",
       "      <td>32279</td>\n",
       "      <td>33149</td>\n",
       "      <td>33179</td>\n",
       "      <td>33243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bowtie2 Aligned to mRNA_Prot %</th>\n",
       "      <td>4.62</td>\n",
       "      <td>5.22</td>\n",
       "      <td>5.22</td>\n",
       "      <td>5.41</td>\n",
       "      <td>5.41</td>\n",
       "      <td>5.40</td>\n",
       "      <td>15.14</td>\n",
       "      <td>16.52</td>\n",
       "      <td>16.52</td>\n",
       "      <td>17.06</td>\n",
       "      <td>...</td>\n",
       "      <td>1.04</td>\n",
       "      <td>1.05</td>\n",
       "      <td>1.05</td>\n",
       "      <td>1.05</td>\n",
       "      <td>.61</td>\n",
       "      <td>.66</td>\n",
       "      <td>.66</td>\n",
       "      <td>.68</td>\n",
       "      <td>.68</td>\n",
       "      <td>.68</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38 rows  528 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                1          2          3    \\\n",
       "0                                                                           \n",
       "sample                                    SFHH005aa  SFHH005aa  SFHH005aa   \n",
       "subject                                         3_5        3_5        3_5   \n",
       "Lab kit                                      D-plex     D-plex     D-plex   \n",
       "trimmer                                      bbduk1     bbduk2     bbduk3   \n",
       "Raw Read Count                              4038879    4038879    4038879   \n",
       "Raw Read Length                             35.0646    35.0646    35.0646   \n",
       "Trimmed Read Count                          1270784    1372973    1372997   \n",
       "Trimmed Ave Read Length                     33.8185    38.2306    38.2314   \n",
       "STAR Aligned to Transcriptome                111104     112499     112492   \n",
       "STAR Aligned to Transcriptome %                8.74       8.19       8.19   \n",
       "STAR Aligned to Genome                       655927     697141     697145   \n",
       "STAR Aligned to Genome %                      51.61      50.77      50.77   \n",
       "STAR Unaligned                               614857     675832     675852   \n",
       "STAR Unaligned %                              48.38      49.22      49.22   \n",
       "STAR Unmapped                                614857     675832     675852   \n",
       "STAR Unmapped %                               48.38      49.22      49.22   \n",
       "Bowtie2 Aligned to hg38 (1)                  165418     204662     204672   \n",
       "Bowtie2 Aligned to hg38 (1) %                 13.01      14.90      14.90   \n",
       "STAR Aligned to mirna                        156554     154884     154884   \n",
       "STAR Aligned to mirna %                       12.31      11.28      11.28   \n",
       "Bowtie Aligned to mirna                       49775      49507      49507   \n",
       "Bowtie Aligned to mirna %                      3.91       3.60       3.60   \n",
       "Bowtie2 Aligned to mirna                        946       1060       1060   \n",
       "Bowtie2 Aligned to mirna %                      .07        .07        .07   \n",
       "Bowtie2 Aligned to RMSK                       92670     116878     116879   \n",
       "Bowtie2 Aligned to RMSK %                      7.29       8.51       8.51   \n",
       "Bowtie2 Aligned to phiX                        7745       9144       9144   \n",
       "Bowtie2 Aligned to phiX %                       .60        .66        .66   \n",
       "Bowtie2 Aligned to Salmonella                 45946      60133      60133   \n",
       "Bowtie2 Aligned to Salmonella %                3.61       4.37       4.37   \n",
       "Bowtie2 Aligned to masked Salmonella          14700      17811      17811   \n",
       "Bowtie2 Aligned to masked Salmonella %         1.15       1.29       1.29   \n",
       "Bowtie2 Aligned to Burkholderia              375543     413414     413414   \n",
       "Bowtie2 Aligned to Burkholderia %             29.55      30.11      30.11   \n",
       "Bowtie2 Aligned to masked Burkholderia       277406     298416     298416   \n",
       "Bowtie2 Aligned to masked Burkholderia %      21.82      21.73      21.73   \n",
       "Bowtie2 Aligned to mRNA_Prot                  58736      71735      71735   \n",
       "Bowtie2 Aligned to mRNA_Prot %                 4.62       5.22       5.22   \n",
       "\n",
       "                                                4          5          6    \\\n",
       "0                                                                           \n",
       "sample                                    SFHH005aa  SFHH005aa  SFHH005aa   \n",
       "subject                                         3_5        3_5        3_5   \n",
       "Lab kit                                      D-plex     D-plex     D-plex   \n",
       "trimmer                                   cutadapt1  cutadapt2  cutadapt3   \n",
       "Raw Read Count                              4038879    4038879    4038879   \n",
       "Raw Read Length                             35.0646    35.0646    35.0646   \n",
       "Trimmed Read Count                          1395891    1395893    1395893   \n",
       "Trimmed Ave Read Length                     43.5121    43.5142     43.524   \n",
       "STAR Aligned to Transcriptome                112689     112696     112689   \n",
       "STAR Aligned to Transcriptome %                8.07       8.07       8.07   \n",
       "STAR Aligned to Genome                       701944     701935     701914   \n",
       "STAR Aligned to Genome %                      50.28      50.28      50.28   \n",
       "STAR Unaligned                               693947     693958     693979   \n",
       "STAR Unaligned %                              49.71      49.71      49.71   \n",
       "STAR Unmapped                                693947     693958     693979   \n",
       "STAR Unmapped %                               49.71      49.71      49.71   \n",
       "Bowtie2 Aligned to hg38 (1)                  208906     208905     208947   \n",
       "Bowtie2 Aligned to hg38 (1) %                 14.96      14.96      14.96   \n",
       "STAR Aligned to mirna                        156555     156554     156554   \n",
       "STAR Aligned to mirna %                       11.21      11.21      11.21   \n",
       "Bowtie Aligned to mirna                       50782      50781      50781   \n",
       "Bowtie Aligned to mirna %                      3.63       3.63       3.63   \n",
       "Bowtie2 Aligned to mirna                       1235       1235       1235   \n",
       "Bowtie2 Aligned to mirna %                      .08        .08        .08   \n",
       "Bowtie2 Aligned to RMSK                      121972     121971     121999   \n",
       "Bowtie2 Aligned to RMSK %                      8.73       8.73       8.73   \n",
       "Bowtie2 Aligned to phiX                       11745      11747      11745   \n",
       "Bowtie2 Aligned to phiX %                       .84        .84        .84   \n",
       "Bowtie2 Aligned to Salmonella                 61284      61285      61282   \n",
       "Bowtie2 Aligned to Salmonella %                4.39       4.39       4.39   \n",
       "Bowtie2 Aligned to masked Salmonella          18218      18218      18215   \n",
       "Bowtie2 Aligned to masked Salmonella %         1.30       1.30       1.30   \n",
       "Bowtie2 Aligned to Burkholderia              420290     420299     420297   \n",
       "Bowtie2 Aligned to Burkholderia %             30.10      30.10      30.10   \n",
       "Bowtie2 Aligned to masked Burkholderia       304330     304339     304338   \n",
       "Bowtie2 Aligned to masked Burkholderia %      21.80      21.80      21.80   \n",
       "Bowtie2 Aligned to mRNA_Prot                  75518      75522      75517   \n",
       "Bowtie2 Aligned to mRNA_Prot %                 5.41       5.41       5.40   \n",
       "\n",
       "                                                7          8          9    \\\n",
       "0                                                                           \n",
       "sample                                    SFHH005ab  SFHH005ab  SFHH005ab   \n",
       "subject                                         3_6        3_6        3_6   \n",
       "Lab kit                                      D-plex     D-plex     D-plex   \n",
       "trimmer                                      bbduk1     bbduk2     bbduk3   \n",
       "Raw Read Count                              3088240    3088240    3088240   \n",
       "Raw Read Length                             48.6946    48.6946    48.6946   \n",
       "Trimmed Read Count                          1240850    1396651    1396740   \n",
       "Trimmed Ave Read Length                     40.8779    51.0991     51.101   \n",
       "STAR Aligned to Transcriptome                 92673      93752      93727   \n",
       "STAR Aligned to Transcriptome %                7.46       6.71       6.71   \n",
       "STAR Aligned to Genome                       727759     834086     834097   \n",
       "STAR Aligned to Genome %                      58.65      59.72      59.71   \n",
       "STAR Unaligned                               513091     562565     562643   \n",
       "STAR Unaligned %                              41.34      40.27      40.28   \n",
       "STAR Unmapped                                513091     562565     562643   \n",
       "STAR Unmapped %                               41.34      40.27      40.28   \n",
       "Bowtie2 Aligned to hg38 (1)                  475443     600590     600639   \n",
       "Bowtie2 Aligned to hg38 (1) %                 38.31      43.00      43.00   \n",
       "STAR Aligned to mirna                        125155     115596     115600   \n",
       "STAR Aligned to mirna %                       10.08       8.27       8.27   \n",
       "Bowtie Aligned to mirna                       39999      36092      36092   \n",
       "Bowtie Aligned to mirna %                      3.22       2.58       2.58   \n",
       "Bowtie2 Aligned to mirna                       4737       5302       5302   \n",
       "Bowtie2 Aligned to mirna %                      .38        .37        .37   \n",
       "Bowtie2 Aligned to RMSK                      304510     388795     388827   \n",
       "Bowtie2 Aligned to RMSK %                     24.54      27.83      27.83   \n",
       "Bowtie2 Aligned to phiX                       40687      47810      47811   \n",
       "Bowtie2 Aligned to phiX %                      3.27       3.42       3.42   \n",
       "Bowtie2 Aligned to Salmonella                 43007      54722      54722   \n",
       "Bowtie2 Aligned to Salmonella %                3.46       3.91       3.91   \n",
       "Bowtie2 Aligned to masked Salmonella          14998      17504      17504   \n",
       "Bowtie2 Aligned to masked Salmonella %         1.20       1.25       1.25   \n",
       "Bowtie2 Aligned to Burkholderia              274240     303638     303638   \n",
       "Bowtie2 Aligned to Burkholderia %             22.10      21.74      21.73   \n",
       "Bowtie2 Aligned to masked Burkholderia       180637     197994     197994   \n",
       "Bowtie2 Aligned to masked Burkholderia %      14.55      14.17      14.17   \n",
       "Bowtie2 Aligned to mRNA_Prot                 187948     230807     230826   \n",
       "Bowtie2 Aligned to mRNA_Prot %                15.14      16.52      16.52   \n",
       "\n",
       "                                                10   ...       519        520  \\\n",
       "0                                                    ...                        \n",
       "sample                                    SFHH005ab  ...  SFHH006y   SFHH006y   \n",
       "subject                                         3_6  ...       3_3        3_3   \n",
       "Lab kit                                      D-plex  ...   Lexogen    Lexogen   \n",
       "trimmer                                   cutadapt1  ...    bbduk3  cutadapt1   \n",
       "Raw Read Count                              3088240  ...   3665258    3665258   \n",
       "Raw Read Length                             48.6946  ...   34.0495    34.0495   \n",
       "Trimmed Read Count                          1418066  ...   3633881    3661993   \n",
       "Trimmed Ave Read Length                     64.3485  ...   32.6867    33.8972   \n",
       "STAR Aligned to Transcriptome                 92285  ...    337860     337076   \n",
       "STAR Aligned to Transcriptome %                6.50  ...      9.29       9.20   \n",
       "STAR Aligned to Genome                       837173  ...   1555401    1547893   \n",
       "STAR Aligned to Genome %                      59.03  ...     42.80      42.26   \n",
       "STAR Unaligned                               580893  ...   2078480    2114100   \n",
       "STAR Unaligned %                              40.96  ...     57.19      57.73   \n",
       "STAR Unmapped                                580893  ...   2078480    2114100   \n",
       "STAR Unmapped %                               40.96  ...     57.19      57.73   \n",
       "Bowtie2 Aligned to hg38 (1)                  609164  ...     46809      57182   \n",
       "Bowtie2 Aligned to hg38 (1) %                 42.95  ...      1.28       1.56   \n",
       "STAR Aligned to mirna                        114309  ...    328284     326524   \n",
       "STAR Aligned to mirna %                        8.06  ...      9.03       8.91   \n",
       "Bowtie Aligned to mirna                       36284  ...     68521      67732   \n",
       "Bowtie Aligned to mirna %                      2.55  ...      1.88       1.84   \n",
       "Bowtie2 Aligned to mirna                       5708  ...     15195      15288   \n",
       "Bowtie2 Aligned to mirna %                      .40  ...       .41        .41   \n",
       "Bowtie2 Aligned to RMSK                      402988  ...     24856      31291   \n",
       "Bowtie2 Aligned to RMSK %                     28.41  ...       .68        .85   \n",
       "Bowtie2 Aligned to phiX                       55340  ...      1303      10874   \n",
       "Bowtie2 Aligned to phiX %                      3.90  ...       .03        .29   \n",
       "Bowtie2 Aligned to Salmonella                 56558  ...     69276      70037   \n",
       "Bowtie2 Aligned to Salmonella %                3.98  ...      1.90       1.91   \n",
       "Bowtie2 Aligned to masked Salmonella          18046  ...     34220      34653   \n",
       "Bowtie2 Aligned to masked Salmonella %         1.27  ...       .94        .94   \n",
       "Bowtie2 Aligned to Burkholderia              308657  ...   2101944    2121875   \n",
       "Bowtie2 Aligned to Burkholderia %             21.76  ...     57.84      57.94   \n",
       "Bowtie2 Aligned to masked Burkholderia       201850  ...   1961758    1980757   \n",
       "Bowtie2 Aligned to masked Burkholderia %      14.23  ...     53.98      54.08   \n",
       "Bowtie2 Aligned to mRNA_Prot                 241994  ...     38124      38784   \n",
       "Bowtie2 Aligned to mRNA_Prot %                17.06  ...      1.04       1.05   \n",
       "\n",
       "                                                521        522       523  \\\n",
       "0                                                                          \n",
       "sample                                     SFHH006y   SFHH006y  SFHH006z   \n",
       "subject                                         3_3        3_3       3_4   \n",
       "Lab kit                                     Lexogen    Lexogen   Lexogen   \n",
       "trimmer                                   cutadapt2  cutadapt3    bbduk1   \n",
       "Raw Read Count                              3665258    3665258   4868169   \n",
       "Raw Read Length                             34.0495    34.0495   36.2413   \n",
       "Trimmed Read Count                          3664115    3664263   4478092   \n",
       "Trimmed Ave Read Length                     34.0071    34.0507   30.6793   \n",
       "STAR Aligned to Transcriptome                336880     336843    362402   \n",
       "STAR Aligned to Transcriptome %                9.19       9.19      8.09   \n",
       "STAR Aligned to Genome                      1547026    1546840   1947074   \n",
       "STAR Aligned to Genome %                      42.22      42.21     43.47   \n",
       "STAR Unaligned                              2117089    2117423   2531018   \n",
       "STAR Unaligned %                              57.77      57.78     56.52   \n",
       "STAR Unmapped                               2117089    2117423   2531018   \n",
       "STAR Unmapped %                               57.77      57.78     56.52   \n",
       "Bowtie2 Aligned to hg38 (1)                   58275      59209     40727   \n",
       "Bowtie2 Aligned to hg38 (1) %                  1.59       1.61       .90   \n",
       "STAR Aligned to mirna                        326446     326417    358302   \n",
       "STAR Aligned to mirna %                        8.90       8.90      8.00   \n",
       "Bowtie Aligned to mirna                       67692      67685     68521   \n",
       "Bowtie Aligned to mirna %                      1.84       1.84      1.53   \n",
       "Bowtie2 Aligned to mirna                      15268      15268     14483   \n",
       "Bowtie2 Aligned to mirna %                      .41        .41       .32   \n",
       "Bowtie2 Aligned to RMSK                       31914      32367     18268   \n",
       "Bowtie2 Aligned to RMSK %                       .87        .88       .40   \n",
       "Bowtie2 Aligned to phiX                       10896      10898      1495   \n",
       "Bowtie2 Aligned to phiX %                       .29        .29       .03   \n",
       "Bowtie2 Aligned to Salmonella                 70056      70047     76595   \n",
       "Bowtie2 Aligned to Salmonella %                1.91       1.91      1.71   \n",
       "Bowtie2 Aligned to masked Salmonella          34689      34684     31307   \n",
       "Bowtie2 Aligned to masked Salmonella %          .94        .94       .69   \n",
       "Bowtie2 Aligned to Burkholderia             2118401    2118297   2622229   \n",
       "Bowtie2 Aligned to Burkholderia %             57.81      57.80     58.55   \n",
       "Bowtie2 Aligned to masked Burkholderia      1977308    1977214   2433977   \n",
       "Bowtie2 Aligned to masked Burkholderia %      53.96      53.95     54.35   \n",
       "Bowtie2 Aligned to mRNA_Prot                  38798      38831     27716   \n",
       "Bowtie2 Aligned to mRNA_Prot %                 1.05       1.05       .61   \n",
       "\n",
       "                                               524       525        526  \\\n",
       "0                                                                         \n",
       "sample                                    SFHH006z  SFHH006z   SFHH006z   \n",
       "subject                                        3_4       3_4        3_4   \n",
       "Lab kit                                    Lexogen   Lexogen    Lexogen   \n",
       "trimmer                                     bbduk2    bbduk3  cutadapt1   \n",
       "Raw Read Count                             4868169   4868169    4868169   \n",
       "Raw Read Length                            36.2413   36.2413    36.2413   \n",
       "Trimmed Read Count                         4823895   4823681    4863961   \n",
       "Trimmed Ave Read Length                    34.7925   34.7983    36.0631   \n",
       "STAR Aligned to Transcriptome               363298    363259     362012   \n",
       "STAR Aligned to Transcriptome %               7.53      7.53       7.44   \n",
       "STAR Aligned to Genome                     1730325   1729961    1718029   \n",
       "STAR Aligned to Genome %                     35.86     35.86      35.32   \n",
       "STAR Unaligned                             3093570   3093720    3145932   \n",
       "STAR Unaligned %                             64.13     64.13      64.67   \n",
       "STAR Unmapped                              3093570   3093720    3145932   \n",
       "STAR Unmapped %                              64.13     64.13      64.67   \n",
       "Bowtie2 Aligned to hg38 (1)                  47736     47746      62246   \n",
       "Bowtie2 Aligned to hg38 (1) %                  .98       .98       1.27   \n",
       "STAR Aligned to mirna                       382995    382987     380420   \n",
       "STAR Aligned to mirna %                       7.93      7.93       7.82   \n",
       "Bowtie Aligned to mirna                      66796     66795      65868   \n",
       "Bowtie Aligned to mirna %                     1.38      1.38       1.35   \n",
       "Bowtie2 Aligned to mirna                     15281     15282      15345   \n",
       "Bowtie2 Aligned to mirna %                     .31       .31        .31   \n",
       "Bowtie2 Aligned to RMSK                      23241     23246      32052   \n",
       "Bowtie2 Aligned to RMSK %                      .48       .48        .65   \n",
       "Bowtie2 Aligned to phiX                       1878      1878      14144   \n",
       "Bowtie2 Aligned to phiX %                      .03       .03        .29   \n",
       "Bowtie2 Aligned to Salmonella               116173    116172     117387   \n",
       "Bowtie2 Aligned to Salmonella %               2.40      2.40       2.41   \n",
       "Bowtie2 Aligned to masked Salmonella         58894     58894      59592   \n",
       "Bowtie2 Aligned to masked Salmonella %        1.22      1.22       1.22   \n",
       "Bowtie2 Aligned to Burkholderia            2985241   2984974    3016302   \n",
       "Bowtie2 Aligned to Burkholderia %            61.88     61.88      62.01   \n",
       "Bowtie2 Aligned to masked Burkholderia     2777831   2777568    2807588   \n",
       "Bowtie2 Aligned to masked Burkholderia %     57.58     57.58      57.72   \n",
       "Bowtie2 Aligned to mRNA_Prot                 32278     32279      33149   \n",
       "Bowtie2 Aligned to mRNA_Prot %                 .66       .66        .68   \n",
       "\n",
       "                                                527        528  \n",
       "0                                                               \n",
       "sample                                     SFHH006z   SFHH006z  \n",
       "subject                                         3_4        3_4  \n",
       "Lab kit                                     Lexogen    Lexogen  \n",
       "trimmer                                   cutadapt2  cutadapt3  \n",
       "Raw Read Count                              4868169    4868169  \n",
       "Raw Read Length                             36.2413    36.2413  \n",
       "Trimmed Read Count                          4867158    4867302  \n",
       "Trimmed Ave Read Length                     36.1974     36.242  \n",
       "STAR Aligned to Transcriptome                361905     361853  \n",
       "STAR Aligned to Transcriptome %                7.43       7.43  \n",
       "STAR Aligned to Genome                      1716972    1716740  \n",
       "STAR Aligned to Genome %                      35.27      35.27  \n",
       "STAR Unaligned                              3150186    3150562  \n",
       "STAR Unaligned %                              64.72      64.72  \n",
       "STAR Unmapped                               3150186    3150562  \n",
       "STAR Unmapped %                               64.72      64.72  \n",
       "Bowtie2 Aligned to hg38 (1)                   63840      65105  \n",
       "Bowtie2 Aligned to hg38 (1) %                  1.31       1.33  \n",
       "STAR Aligned to mirna                        380500     380475  \n",
       "STAR Aligned to mirna %                        7.81       7.81  \n",
       "Bowtie Aligned to mirna                       65889      65886  \n",
       "Bowtie Aligned to mirna %                      1.35       1.35  \n",
       "Bowtie2 Aligned to mirna                      15333      15333  \n",
       "Bowtie2 Aligned to mirna %                      .31        .31  \n",
       "Bowtie2 Aligned to RMSK                       32955      33561  \n",
       "Bowtie2 Aligned to RMSK %                       .67        .68  \n",
       "Bowtie2 Aligned to phiX                       14180      14182  \n",
       "Bowtie2 Aligned to phiX %                       .29        .29  \n",
       "Bowtie2 Aligned to Salmonella                117383     117374  \n",
       "Bowtie2 Aligned to Salmonella %                2.41       2.41  \n",
       "Bowtie2 Aligned to masked Salmonella          59593      59589  \n",
       "Bowtie2 Aligned to masked Salmonella %         1.22       1.22  \n",
       "Bowtie2 Aligned to Burkholderia             3009621    3009486  \n",
       "Bowtie2 Aligned to Burkholderia %             61.83      61.83  \n",
       "Bowtie2 Aligned to masked Burkholderia      2800958    2800836  \n",
       "Bowtie2 Aligned to masked Burkholderia %      57.54      57.54  \n",
       "Bowtie2 Aligned to mRNA_Prot                  33179      33243  \n",
       "Bowtie2 Aligned to mRNA_Prot %                  .68        .68  \n",
       "\n",
       "[38 rows x 528 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>subject</th>\n",
       "      <th>Lab kit</th>\n",
       "      <th>trimmer</th>\n",
       "      <th>Raw Read Count</th>\n",
       "      <th>Raw Read Length</th>\n",
       "      <th>Trimmed Read Count</th>\n",
       "      <th>Trimmed Ave Read Length</th>\n",
       "      <th>STAR Aligned to Transcriptome</th>\n",
       "      <th>STAR Aligned to Transcriptome %</th>\n",
       "      <th>...</th>\n",
       "      <th>Bowtie2 Aligned to Salmonella</th>\n",
       "      <th>Bowtie2 Aligned to Salmonella %</th>\n",
       "      <th>Bowtie2 Aligned to masked Salmonella</th>\n",
       "      <th>Bowtie2 Aligned to masked Salmonella %</th>\n",
       "      <th>Bowtie2 Aligned to Burkholderia</th>\n",
       "      <th>Bowtie2 Aligned to Burkholderia %</th>\n",
       "      <th>Bowtie2 Aligned to masked Burkholderia</th>\n",
       "      <th>Bowtie2 Aligned to masked Burkholderia %</th>\n",
       "      <th>Bowtie2 Aligned to mRNA_Prot</th>\n",
       "      <th>Bowtie2 Aligned to mRNA_Prot %</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1270784</td>\n",
       "      <td>33.8185</td>\n",
       "      <td>111104</td>\n",
       "      <td>8.74</td>\n",
       "      <td>...</td>\n",
       "      <td>45946</td>\n",
       "      <td>3.61</td>\n",
       "      <td>14700</td>\n",
       "      <td>1.15</td>\n",
       "      <td>375543</td>\n",
       "      <td>29.55</td>\n",
       "      <td>277406</td>\n",
       "      <td>21.82</td>\n",
       "      <td>58736</td>\n",
       "      <td>4.62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372973</td>\n",
       "      <td>38.2306</td>\n",
       "      <td>112499</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>60133</td>\n",
       "      <td>4.37</td>\n",
       "      <td>17811</td>\n",
       "      <td>1.29</td>\n",
       "      <td>413414</td>\n",
       "      <td>30.11</td>\n",
       "      <td>298416</td>\n",
       "      <td>21.73</td>\n",
       "      <td>71735</td>\n",
       "      <td>5.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372997</td>\n",
       "      <td>38.2314</td>\n",
       "      <td>112492</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>60133</td>\n",
       "      <td>4.37</td>\n",
       "      <td>17811</td>\n",
       "      <td>1.29</td>\n",
       "      <td>413414</td>\n",
       "      <td>30.11</td>\n",
       "      <td>298416</td>\n",
       "      <td>21.73</td>\n",
       "      <td>71735</td>\n",
       "      <td>5.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395891</td>\n",
       "      <td>43.5121</td>\n",
       "      <td>112689</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>61284</td>\n",
       "      <td>4.39</td>\n",
       "      <td>18218</td>\n",
       "      <td>1.30</td>\n",
       "      <td>420290</td>\n",
       "      <td>30.10</td>\n",
       "      <td>304330</td>\n",
       "      <td>21.80</td>\n",
       "      <td>75518</td>\n",
       "      <td>5.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395893</td>\n",
       "      <td>43.5142</td>\n",
       "      <td>112696</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>61285</td>\n",
       "      <td>4.39</td>\n",
       "      <td>18218</td>\n",
       "      <td>1.30</td>\n",
       "      <td>420299</td>\n",
       "      <td>30.10</td>\n",
       "      <td>304339</td>\n",
       "      <td>21.80</td>\n",
       "      <td>75522</td>\n",
       "      <td>5.41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0     sample subject Lab kit    trimmer Raw Read Count Raw Read Length  \\\n",
       "1  SFHH005aa     3_5  D-plex     bbduk1        4038879         35.0646   \n",
       "2  SFHH005aa     3_5  D-plex     bbduk2        4038879         35.0646   \n",
       "3  SFHH005aa     3_5  D-plex     bbduk3        4038879         35.0646   \n",
       "4  SFHH005aa     3_5  D-plex  cutadapt1        4038879         35.0646   \n",
       "5  SFHH005aa     3_5  D-plex  cutadapt2        4038879         35.0646   \n",
       "\n",
       "0 Trimmed Read Count Trimmed Ave Read Length STAR Aligned to Transcriptome  \\\n",
       "1            1270784                 33.8185                        111104   \n",
       "2            1372973                 38.2306                        112499   \n",
       "3            1372997                 38.2314                        112492   \n",
       "4            1395891                 43.5121                        112689   \n",
       "5            1395893                 43.5142                        112696   \n",
       "\n",
       "0 STAR Aligned to Transcriptome %  ... Bowtie2 Aligned to Salmonella  \\\n",
       "1                            8.74  ...                         45946   \n",
       "2                            8.19  ...                         60133   \n",
       "3                            8.19  ...                         60133   \n",
       "4                            8.07  ...                         61284   \n",
       "5                            8.07  ...                         61285   \n",
       "\n",
       "0 Bowtie2 Aligned to Salmonella % Bowtie2 Aligned to masked Salmonella  \\\n",
       "1                            3.61                                14700   \n",
       "2                            4.37                                17811   \n",
       "3                            4.37                                17811   \n",
       "4                            4.39                                18218   \n",
       "5                            4.39                                18218   \n",
       "\n",
       "0 Bowtie2 Aligned to masked Salmonella % Bowtie2 Aligned to Burkholderia  \\\n",
       "1                                   1.15                          375543   \n",
       "2                                   1.29                          413414   \n",
       "3                                   1.29                          413414   \n",
       "4                                   1.30                          420290   \n",
       "5                                   1.30                          420299   \n",
       "\n",
       "0 Bowtie2 Aligned to Burkholderia % Bowtie2 Aligned to masked Burkholderia  \\\n",
       "1                             29.55                                 277406   \n",
       "2                             30.11                                 298416   \n",
       "3                             30.11                                 298416   \n",
       "4                             30.10                                 304330   \n",
       "5                             30.10                                 304339   \n",
       "\n",
       "0 Bowtie2 Aligned to masked Burkholderia % Bowtie2 Aligned to mRNA_Prot  \\\n",
       "1                                    21.82                        58736   \n",
       "2                                    21.73                        71735   \n",
       "3                                    21.73                        71735   \n",
       "4                                    21.80                        75518   \n",
       "5                                    21.80                        75522   \n",
       "\n",
       "0 Bowtie2 Aligned to mRNA_Prot %  \n",
       "1                           4.62  \n",
       "2                           5.22  \n",
       "3                           5.22  \n",
       "4                           5.41  \n",
       "5                           5.41  \n",
       "\n",
       "[5 rows x 38 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=df.T\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>subject</th>\n",
       "      <th>Lab kit</th>\n",
       "      <th>trimmer</th>\n",
       "      <th>Raw Read Count</th>\n",
       "      <th>Raw Read Length</th>\n",
       "      <th>Trimmed Read Count</th>\n",
       "      <th>Trimmed Ave Read Length</th>\n",
       "      <th>STAR Aligned to Transcriptome</th>\n",
       "      <th>STAR Aligned to Transcriptome %</th>\n",
       "      <th>...</th>\n",
       "      <th>Bowtie2 Aligned to masked Burkholderia</th>\n",
       "      <th>Bowtie2 Aligned to masked Burkholderia %</th>\n",
       "      <th>Bowtie2 Aligned to mRNA_Prot</th>\n",
       "      <th>Bowtie2 Aligned to mRNA_Prot %</th>\n",
       "      <th>dplex</th>\n",
       "      <th>lexogen</th>\n",
       "      <th>diag</th>\n",
       "      <th>disease</th>\n",
       "      <th>idh</th>\n",
       "      <th>x1p19q</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1270784</td>\n",
       "      <td>33.8185</td>\n",
       "      <td>111104</td>\n",
       "      <td>8.74</td>\n",
       "      <td>...</td>\n",
       "      <td>277406</td>\n",
       "      <td>21.82</td>\n",
       "      <td>58736</td>\n",
       "      <td>4.62</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372973</td>\n",
       "      <td>38.2306</td>\n",
       "      <td>112499</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>298416</td>\n",
       "      <td>21.73</td>\n",
       "      <td>71735</td>\n",
       "      <td>5.22</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372997</td>\n",
       "      <td>38.2314</td>\n",
       "      <td>112492</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>298416</td>\n",
       "      <td>21.73</td>\n",
       "      <td>71735</td>\n",
       "      <td>5.22</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395891</td>\n",
       "      <td>43.5121</td>\n",
       "      <td>112689</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>304330</td>\n",
       "      <td>21.80</td>\n",
       "      <td>75518</td>\n",
       "      <td>5.41</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395893</td>\n",
       "      <td>43.5142</td>\n",
       "      <td>112696</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>304339</td>\n",
       "      <td>21.80</td>\n",
       "      <td>75522</td>\n",
       "      <td>5.41</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>SFHH006aa</td>\n",
       "      <td>GBM, IDH-mutant,</td>\n",
       "      <td>GBM</td>\n",
       "      <td>IDH-mutant</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample subject Lab kit    trimmer Raw Read Count Raw Read Length  \\\n",
       "0  SFHH005aa     3_5  D-plex     bbduk1        4038879         35.0646   \n",
       "1  SFHH005aa     3_5  D-plex     bbduk2        4038879         35.0646   \n",
       "2  SFHH005aa     3_5  D-plex     bbduk3        4038879         35.0646   \n",
       "3  SFHH005aa     3_5  D-plex  cutadapt1        4038879         35.0646   \n",
       "4  SFHH005aa     3_5  D-plex  cutadapt2        4038879         35.0646   \n",
       "\n",
       "  Trimmed Read Count Trimmed Ave Read Length STAR Aligned to Transcriptome  \\\n",
       "0            1270784                 33.8185                        111104   \n",
       "1            1372973                 38.2306                        112499   \n",
       "2            1372997                 38.2314                        112492   \n",
       "3            1395891                 43.5121                        112689   \n",
       "4            1395893                 43.5142                        112696   \n",
       "\n",
       "  STAR Aligned to Transcriptome %  ... Bowtie2 Aligned to masked Burkholderia  \\\n",
       "0                            8.74  ...                                 277406   \n",
       "1                            8.19  ...                                 298416   \n",
       "2                            8.19  ...                                 298416   \n",
       "3                            8.07  ...                                 304330   \n",
       "4                            8.07  ...                                 304339   \n",
       "\n",
       "  Bowtie2 Aligned to masked Burkholderia % Bowtie2 Aligned to mRNA_Prot  \\\n",
       "0                                    21.82                        58736   \n",
       "1                                    21.73                        71735   \n",
       "2                                    21.73                        71735   \n",
       "3                                    21.80                        75518   \n",
       "4                                    21.80                        75522   \n",
       "\n",
       "  Bowtie2 Aligned to mRNA_Prot %      dplex    lexogen              diag  \\\n",
       "0                           4.62  SFHH005aa  SFHH006aa  GBM, IDH-mutant,   \n",
       "1                           5.22  SFHH005aa  SFHH006aa  GBM, IDH-mutant,   \n",
       "2                           5.22  SFHH005aa  SFHH006aa  GBM, IDH-mutant,   \n",
       "3                           5.41  SFHH005aa  SFHH006aa  GBM, IDH-mutant,   \n",
       "4                           5.41  SFHH005aa  SFHH006aa  GBM, IDH-mutant,   \n",
       "\n",
       "  disease          idh x1p19q  \n",
       "0     GBM   IDH-mutant         \n",
       "1     GBM   IDH-mutant         \n",
       "2     GBM   IDH-mutant         \n",
       "3     GBM   IDH-mutant         \n",
       "4     GBM   IDH-mutant         \n",
       "\n",
       "[5 rows x 44 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df=df.merge(meta,left_on='Subject',right_on='subject')\n",
    "df=df.merge(meta,on='subject')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SFHH005aa.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005aa.bbduk2.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005aa.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005aa.cutadapt1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005aa.cutadapt2.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005aa.cutadapt3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005ab.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005ab.bbduk2.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005ab.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH005a.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>SFHH006y.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006y.cutadapt1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006y.cutadapt2.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006y.cutadapt3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006z.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006z.bbduk2.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006z.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006z.cutadapt1.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006z.cutadapt2.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "      <th>SFHH006z.cutadapt3.bowtie2.mRNA_Prot.bam.mrna_counts</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sequence</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NM_000014.6</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NM_000015.3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NM_000016.6</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NM_000020.3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NM_000021.4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  528 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             SFHH005aa.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                          \n",
       "NM_000014.6                                                  0    \n",
       "NM_000015.3                                                  0    \n",
       "NM_000016.6                                                  0    \n",
       "NM_000020.3                                                  0    \n",
       "NM_000021.4                                                  0    \n",
       "\n",
       "             SFHH005aa.bbduk2.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                          \n",
       "NM_000014.6                                                  0    \n",
       "NM_000015.3                                                  0    \n",
       "NM_000016.6                                                  0    \n",
       "NM_000020.3                                                  0    \n",
       "NM_000021.4                                                  0    \n",
       "\n",
       "             SFHH005aa.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                          \n",
       "NM_000014.6                                                  0    \n",
       "NM_000015.3                                                  0    \n",
       "NM_000016.6                                                  0    \n",
       "NM_000020.3                                                  0    \n",
       "NM_000021.4                                                  0    \n",
       "\n",
       "             SFHH005aa.cutadapt1.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                             \n",
       "NM_000014.6                                                  0       \n",
       "NM_000015.3                                                  0       \n",
       "NM_000016.6                                                  0       \n",
       "NM_000020.3                                                  0       \n",
       "NM_000021.4                                                  0       \n",
       "\n",
       "             SFHH005aa.cutadapt2.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                             \n",
       "NM_000014.6                                                  0       \n",
       "NM_000015.3                                                  0       \n",
       "NM_000016.6                                                  0       \n",
       "NM_000020.3                                                  0       \n",
       "NM_000021.4                                                  0       \n",
       "\n",
       "             SFHH005aa.cutadapt3.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                             \n",
       "NM_000014.6                                                  0       \n",
       "NM_000015.3                                                  0       \n",
       "NM_000016.6                                                  0       \n",
       "NM_000020.3                                                  0       \n",
       "NM_000021.4                                                  0       \n",
       "\n",
       "             SFHH005ab.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                          \n",
       "NM_000014.6                                                  3    \n",
       "NM_000015.3                                                  0    \n",
       "NM_000016.6                                                  0    \n",
       "NM_000020.3                                                  1    \n",
       "NM_000021.4                                                  1    \n",
       "\n",
       "             SFHH005ab.bbduk2.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                          \n",
       "NM_000014.6                                                  6    \n",
       "NM_000015.3                                                  0    \n",
       "NM_000016.6                                                  0    \n",
       "NM_000020.3                                                  0    \n",
       "NM_000021.4                                                  1    \n",
       "\n",
       "             SFHH005ab.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                          \n",
       "NM_000014.6                                                  6    \n",
       "NM_000015.3                                                  0    \n",
       "NM_000016.6                                                  0    \n",
       "NM_000020.3                                                  0    \n",
       "NM_000021.4                                                  1    \n",
       "\n",
       "             SFHH005a.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts  ...  \\\n",
       "sequence                                                        ...   \n",
       "NM_000014.6                                                  0  ...   \n",
       "NM_000015.3                                                  0  ...   \n",
       "NM_000016.6                                                  0  ...   \n",
       "NM_000020.3                                                  1  ...   \n",
       "NM_000021.4                                                  2  ...   \n",
       "\n",
       "             SFHH006y.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                         \n",
       "NM_000014.6                                                  0   \n",
       "NM_000015.3                                                  0   \n",
       "NM_000016.6                                                  0   \n",
       "NM_000020.3                                                  0   \n",
       "NM_000021.4                                                  0   \n",
       "\n",
       "             SFHH006y.cutadapt1.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                            \n",
       "NM_000014.6                                                  0      \n",
       "NM_000015.3                                                  0      \n",
       "NM_000016.6                                                  0      \n",
       "NM_000020.3                                                  0      \n",
       "NM_000021.4                                                  0      \n",
       "\n",
       "             SFHH006y.cutadapt2.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                            \n",
       "NM_000014.6                                                  0      \n",
       "NM_000015.3                                                  0      \n",
       "NM_000016.6                                                  0      \n",
       "NM_000020.3                                                  0      \n",
       "NM_000021.4                                                  0      \n",
       "\n",
       "             SFHH006y.cutadapt3.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                            \n",
       "NM_000014.6                                                  0      \n",
       "NM_000015.3                                                  0      \n",
       "NM_000016.6                                                  0      \n",
       "NM_000020.3                                                  0      \n",
       "NM_000021.4                                                  0      \n",
       "\n",
       "             SFHH006z.bbduk1.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                         \n",
       "NM_000014.6                                                  0   \n",
       "NM_000015.3                                                  0   \n",
       "NM_000016.6                                                  0   \n",
       "NM_000020.3                                                  0   \n",
       "NM_000021.4                                                  0   \n",
       "\n",
       "             SFHH006z.bbduk2.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                         \n",
       "NM_000014.6                                                  0   \n",
       "NM_000015.3                                                  0   \n",
       "NM_000016.6                                                  0   \n",
       "NM_000020.3                                                  0   \n",
       "NM_000021.4                                                  0   \n",
       "\n",
       "             SFHH006z.bbduk3.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                         \n",
       "NM_000014.6                                                  0   \n",
       "NM_000015.3                                                  0   \n",
       "NM_000016.6                                                  0   \n",
       "NM_000020.3                                                  0   \n",
       "NM_000021.4                                                  0   \n",
       "\n",
       "             SFHH006z.cutadapt1.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                            \n",
       "NM_000014.6                                                  0      \n",
       "NM_000015.3                                                  0      \n",
       "NM_000016.6                                                  0      \n",
       "NM_000020.3                                                  0      \n",
       "NM_000021.4                                                  0      \n",
       "\n",
       "             SFHH006z.cutadapt2.bowtie2.mRNA_Prot.bam.mrna_counts  \\\n",
       "sequence                                                            \n",
       "NM_000014.6                                                  0      \n",
       "NM_000015.3                                                  0      \n",
       "NM_000016.6                                                  0      \n",
       "NM_000020.3                                                  0      \n",
       "NM_000021.4                                                  0      \n",
       "\n",
       "             SFHH006z.cutadapt3.bowtie2.mRNA_Prot.bam.mrna_counts  \n",
       "sequence                                                           \n",
       "NM_000014.6                                                  0     \n",
       "NM_000015.3                                                  0     \n",
       "NM_000016.6                                                  0     \n",
       "NM_000020.3                                                  0     \n",
       "NM_000021.4                                                  0     \n",
       "\n",
       "[5 rows x 528 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#element_counts = pd.read_csv('20210428-EV/rmsk_class_counts.csv.gz',index_col=0,low_memory=False)\n",
    "#element_counts = pd.read_csv('20210428-EV/rmsk_family_counts.csv.gz',index_col=0,low_memory=False)\n",
    "#element_counts = pd.read_csv('20210428-EV/rmsk_name_counts.csv.gz',index_col=0,low_memory=False)\n",
    "#element_counts = pd.read_csv('20210428-EV/mirna_counts.csv.gz',index_col=0,low_memory=False)\n",
    "#element_counts = pd.read_csv('20210428-EV/gene_counts.csv.gz',index_col=0,low_memory=False)\n",
    "element_counts = pd.read_csv('20210428-EV/mrna_counts.csv.gz',index_col=0,low_memory=False)\n",
    "# diamond_family_counts\n",
    "\n",
    "element_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Keep less than 5000 columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "124152"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(element_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "#element_counts=element_counts[0:5000]\n",
    "#element_counts=element_counts[5000:10000]\n",
    "#element_counts=element_counts[10000:15000]\n",
    "#element_counts=element_counts[15000:20000]\n",
    "#element_counts=element_counts[20000:25000]\n",
    "#element_counts=element_counts[25000:30000]\n",
    "#element_counts=element_counts[30000:35000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "124152"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(element_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>sequence</th>\n",
       "      <th>NM_000014.6</th>\n",
       "      <th>NM_000015.3</th>\n",
       "      <th>NM_000016.6</th>\n",
       "      <th>NM_000020.3</th>\n",
       "      <th>NM_000021.4</th>\n",
       "      <th>NM_000022.4</th>\n",
       "      <th>NM_000025.3</th>\n",
       "      <th>NM_000026.4</th>\n",
       "      <th>NM_000027.4</th>\n",
       "      <th>NM_000028.2</th>\n",
       "      <th>...</th>\n",
       "      <th>XR_953250.2</th>\n",
       "      <th>XR_953252.2</th>\n",
       "      <th>XR_953253.2</th>\n",
       "      <th>XR_953316.2</th>\n",
       "      <th>XR_953317.3</th>\n",
       "      <th>XR_953318.3</th>\n",
       "      <th>XR_953324.3</th>\n",
       "      <th>XR_953325.3</th>\n",
       "      <th>sample</th>\n",
       "      <th>trimmer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>bbduk1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>bbduk2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>bbduk3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>cutadapt1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>cutadapt2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  124154 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "sequence  NM_000014.6  NM_000015.3  NM_000016.6  NM_000020.3  NM_000021.4  \\\n",
       "0                   0            0            0            0            0   \n",
       "1                   0            0            0            0            0   \n",
       "2                   0            0            0            0            0   \n",
       "3                   0            0            0            0            0   \n",
       "4                   0            0            0            0            0   \n",
       "\n",
       "sequence  NM_000022.4  NM_000025.3  NM_000026.4  NM_000027.4  NM_000028.2  \\\n",
       "0                   0            0            1            1            0   \n",
       "1                   0            0            0            1            0   \n",
       "2                   0            0            0            1            0   \n",
       "3                   0            0            1            0            0   \n",
       "4                   0            0            1            0            0   \n",
       "\n",
       "sequence  ...  XR_953250.2  XR_953252.2  XR_953253.2  XR_953316.2  \\\n",
       "0         ...            0            0            0            0   \n",
       "1         ...            0            0            0            0   \n",
       "2         ...            0            0            0            0   \n",
       "3         ...            0            0            0            0   \n",
       "4         ...            0            0            0            0   \n",
       "\n",
       "sequence  XR_953317.3  XR_953318.3  XR_953324.3  XR_953325.3     sample  \\\n",
       "0                   1            0            0            0  SFHH005aa   \n",
       "1                   1            0            0            0  SFHH005aa   \n",
       "2                   1            0            0            0  SFHH005aa   \n",
       "3                   1            0            0            0  SFHH005aa   \n",
       "4                   1            0            0            0  SFHH005aa   \n",
       "\n",
       "sequence    trimmer  \n",
       "0            bbduk1  \n",
       "1            bbduk2  \n",
       "2            bbduk3  \n",
       "3         cutadapt1  \n",
       "4         cutadapt2  \n",
       "\n",
       "[5 rows x 124154 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "element_counts=element_counts.T\n",
    "element_counts=element_counts.reset_index()\n",
    "element_counts[['sample','trimmer','other']]=element_counts['index'].str.split(\".\",2,expand=True)\n",
    "element_counts.drop([\"index\",\"other\"],inplace=True,axis='columns')\n",
    "element_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(528, 124196)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=df.merge(element_counts,on=['sample','trimmer'])\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>subject</th>\n",
       "      <th>Lab kit</th>\n",
       "      <th>trimmer</th>\n",
       "      <th>Raw Read Count</th>\n",
       "      <th>Raw Read Length</th>\n",
       "      <th>Trimmed Read Count</th>\n",
       "      <th>Trimmed Ave Read Length</th>\n",
       "      <th>STAR Aligned to Transcriptome</th>\n",
       "      <th>STAR Aligned to Transcriptome %</th>\n",
       "      <th>...</th>\n",
       "      <th>XR_953246.2</th>\n",
       "      <th>XR_953247.2</th>\n",
       "      <th>XR_953250.2</th>\n",
       "      <th>XR_953252.2</th>\n",
       "      <th>XR_953253.2</th>\n",
       "      <th>XR_953316.2</th>\n",
       "      <th>XR_953317.3</th>\n",
       "      <th>XR_953318.3</th>\n",
       "      <th>XR_953324.3</th>\n",
       "      <th>XR_953325.3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1270784</td>\n",
       "      <td>33.8185</td>\n",
       "      <td>111104</td>\n",
       "      <td>8.74</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372973</td>\n",
       "      <td>38.2306</td>\n",
       "      <td>112499</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372997</td>\n",
       "      <td>38.2314</td>\n",
       "      <td>112492</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395891</td>\n",
       "      <td>43.5121</td>\n",
       "      <td>112689</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395893</td>\n",
       "      <td>43.5142</td>\n",
       "      <td>112696</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  124196 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample subject Lab kit    trimmer Raw Read Count Raw Read Length  \\\n",
       "0  SFHH005aa     3_5  D-plex     bbduk1        4038879         35.0646   \n",
       "1  SFHH005aa     3_5  D-plex     bbduk2        4038879         35.0646   \n",
       "2  SFHH005aa     3_5  D-plex     bbduk3        4038879         35.0646   \n",
       "3  SFHH005aa     3_5  D-plex  cutadapt1        4038879         35.0646   \n",
       "4  SFHH005aa     3_5  D-plex  cutadapt2        4038879         35.0646   \n",
       "\n",
       "  Trimmed Read Count Trimmed Ave Read Length STAR Aligned to Transcriptome  \\\n",
       "0            1270784                 33.8185                        111104   \n",
       "1            1372973                 38.2306                        112499   \n",
       "2            1372997                 38.2314                        112492   \n",
       "3            1395891                 43.5121                        112689   \n",
       "4            1395893                 43.5142                        112696   \n",
       "\n",
       "  STAR Aligned to Transcriptome %  ... XR_953246.2 XR_953247.2 XR_953250.2  \\\n",
       "0                            8.74  ...           0           0           0   \n",
       "1                            8.19  ...           0           0           0   \n",
       "2                            8.19  ...           0           0           0   \n",
       "3                            8.07  ...           0           0           0   \n",
       "4                            8.07  ...           0           0           0   \n",
       "\n",
       "  XR_953252.2 XR_953253.2 XR_953316.2 XR_953317.3 XR_953318.3 XR_953324.3  \\\n",
       "0           0           0           0           1           0           0   \n",
       "1           0           0           0           1           0           0   \n",
       "2           0           0           0           1           0           0   \n",
       "3           0           0           0           1           0           0   \n",
       "4           0           0           0           1           0           0   \n",
       "\n",
       "  XR_953325.3  \n",
       "0           0  \n",
       "1           0  \n",
       "2           0  \n",
       "3           0  \n",
       "4           0  \n",
       "\n",
       "[5 rows x 124196 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['sample', 'subject', 'Lab kit', 'trimmer', 'Raw Read Count',\n",
       "       'Raw Read Length', 'Trimmed Read Count', 'Trimmed Ave Read Length',\n",
       "       'STAR Aligned to Transcriptome', 'STAR Aligned to Transcriptome %',\n",
       "       ...\n",
       "       'XR_953246.2', 'XR_953247.2', 'XR_953250.2', 'XR_953252.2',\n",
       "       'XR_953253.2', 'XR_953316.2', 'XR_953317.3', 'XR_953318.3',\n",
       "       'XR_953324.3', 'XR_953325.3'],\n",
       "      dtype='object', length=124196)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns  #.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['NM_000014.6', 'NM_000015.3', 'NM_000016.6', 'NM_000020.3',\n",
       "       'NM_000021.4', 'NM_000022.4', 'NM_000025.3', 'NM_000026.4',\n",
       "       'NM_000027.4', 'NM_000028.2',\n",
       "       ...\n",
       "       'XR_953246.2', 'XR_953247.2', 'XR_953250.2', 'XR_953252.2',\n",
       "       'XR_953253.2', 'XR_953316.2', 'XR_953317.3', 'XR_953318.3',\n",
       "       'XR_953324.3', 'XR_953325.3'],\n",
       "      dtype='object', name='sequence', length=124152)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#genes=df.columns[92:142]\n",
    "#genes=df.columns[100:554]\n",
    "#a=df.columns.tolist().index('STAR Gene Counts')+1\n",
    "#b=df.columns.tolist().index('STAR miRNA Counts')\n",
    "#genes=df.columns[a:b]\n",
    "elements=element_counts.columns[:-2]\n",
    "elements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['Raw Read Count'] = pd.to_numeric(df['Raw Read Count'], errors='coerce')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['Trimmed Read Count'] = pd.to_numeric(df['Trimmed Read Count'], errors='coerce')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows',None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#df[['subject','sample','Raw Read Count']].drop_duplicates().sort_values(by='Raw Read Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#df[['subject','sample','trimmer','Trimmed Read Count']].drop_duplicates().sort_values(by='Trimmed Read Count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows',default_max_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepping to ttest all items against all other items"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['GBM, IDH-mutant,', 'Diffuse Astrocytoma, IDH-mutant,',\n",
       "       'Oligodendroglioma, IDH-mutant, 1p19q codeleted',\n",
       "       'V01 control (S1),,', 'GBM, IDH1R132H WT,', 'blank 2 (C1),,',\n",
       "       'blank1 (C1),,'], dtype=object)"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['diag'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "u=df['diag'].unique()\n",
    "diags=u[0:3]\n",
    "diags=np.append(diags,u[4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['GBM, IDH-mutant,', 'Diffuse Astrocytoma, IDH-mutant,',\n",
       "       'Oligodendroglioma, IDH-mutant, 1p19q codeleted',\n",
       "       'GBM, IDH1R132H WT,'], dtype=object)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "numpy.ndarray"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(diags)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "diags=diags.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GBM, IDH-mutant,',\n",
       " 'Diffuse Astrocytoma, IDH-mutant,',\n",
       " 'Oligodendroglioma, IDH-mutant, 1p19q codeleted',\n",
       " 'GBM, IDH1R132H WT,']"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Why was this here?\n",
    "\n",
    "#import warnings\n",
    "#warnings.filterwarnings(\"ignore\", category=np.VisibleDeprecationWarning) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "pca plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "def ttests_boxplots_and_heatmaps(localdf,columns,box_p=0.05,heat_p=0.15):\n",
    "    select_p_values=[]\n",
    "    for col in columns:\n",
    "\n",
    "        if ( col not in localdf.columns ):\n",
    "            continue\n",
    "\n",
    "        for labkit in localdf['Lab kit'].unique():\n",
    "            #\"D-plex\",\"Lexogen\":\n",
    "            for trimmer in 'bbduk2','cutadapt2':\n",
    "                #localdf['Trimmer'].unique():\n",
    "                #\"bbduk1\",\"bbduk2\",\"bbduk3\",\"cutadapt1\",\"cutadapt2\",\"cutadapt3\":\n",
    "                for diag in diags:\n",
    "                    others=diags.copy()\n",
    "                    others.remove(diag)\n",
    "                    for other in others:\n",
    "                        t, p = stats.ttest_ind(\n",
    "                            localdf[((localdf[\"Lab kit\"]==labkit) &\n",
    "                                     (localdf[\"trimmer\"]==trimmer) & \n",
    "                                (localdf[\"diag\"]==diag))][col],\n",
    "                            localdf[((localdf[\"Lab kit\"]==labkit) &\n",
    "                                     (localdf[\"trimmer\"]==trimmer) & \n",
    "                                (localdf[\"diag\"]==other))][col])\n",
    "                        if p < heat_p:\n",
    "                            select_p_values.append([abs(t),p,col,labkit,trimmer])\n",
    "\n",
    "    pdf = pd.DataFrame(select_p_values, columns=['t','p','col','labkit','trimmer'])                        \n",
    "    pdf=pdf.drop_duplicates().sort_values('p')\n",
    "\n",
    "    for index, row in pdf.iterrows():\n",
    "        if row['p'] < box_p:\n",
    "            print(\" p : \"+str(row['p'])+\"  ( t : \"+str(row['t'])+\" ) :  \"+\n",
    "                  row['labkit']+\"  :  \"+row['trimmer']+\"  :  \"+row['col'])\n",
    "            print(\"Control and blanks\")\n",
    "            print(localdf[\n",
    "                (localdf['diag']=='V01 control (S1),,') &\n",
    "                (localdf[\"Lab kit\"]==row['labkit']) &\n",
    "                (localdf[\"trimmer\"]==row['trimmer'])][row['col']])\n",
    "            print(localdf[\n",
    "                (localdf['diag']=='blank 2 (C1),,') &\n",
    "                (localdf[\"Lab kit\"]==row['labkit']) &\n",
    "                (localdf[\"trimmer\"]==row['trimmer'])][row['col']])\n",
    "            print(localdf[\n",
    "                (localdf['diag']=='blank1 (C1),,') &\n",
    "                (localdf[\"Lab kit\"]==row['labkit']) &\n",
    "                (localdf[\"trimmer\"]==row['trimmer'])][row['col']])\n",
    "                        \n",
    "            localdf[((localdf[\"Lab kit\"]==row['labkit']) &\n",
    "                     (localdf[\"trimmer\"]==row['trimmer']))].boxplot(\n",
    "                column=row['col'],by=['diag'])\n",
    "            plt.title(row['col'] + \" Normalized by Trimmed Read Count\")\n",
    "            plt.xticks(rotation=75,ha='right')\n",
    "            plt.show()\n",
    "\n",
    "    selected=['subject','sample','Lab kit','trimmer','diag','idh']\n",
    "    selected=np.append(selected,pdf['col'])\n",
    "    selected=np.unique(selected)\n",
    "\n",
    "    #['Lab kit' 'NR_029609.1' 'trimmer' 'diag' 'idh' 'sample' 'subject']\n",
    "    if len(selected) > 7:\n",
    "        for trimmer in 'bbduk2','cutadapt2':\n",
    "            for labkit in dfn['Lab kit'].unique():\n",
    "                tmp=localdf[selected].copy()\n",
    "                tmp=tmp[(tmp['trimmer']==trimmer) & (tmp['Lab kit']==labkit)]\n",
    "                tmp=tmp.drop('trimmer',axis='columns')\n",
    "                tmp=tmp.drop('Lab kit',axis='columns')\n",
    "                tmp.set_index(['subject','sample','diag','idh'],inplace=True)\n",
    "                \n",
    "                #tmp-=tmp.min() # This may almost always be 0 now.\n",
    "                tmp/=tmp.max() # given that there should be no NAs now, coule let clustermap do this\n",
    "                # ...\n",
    "                #standard_scale int or None, optional\n",
    "                #Either 0 (rows) or 1 (columns).\n",
    "                #Whether or not to standardize that dimension, meaning for each row or column,\n",
    "                #subtract the minimum and divide each by its maximum.\n",
    "                # how would that deal with 0s or empty cells\n",
    "\n",
    "                tmp=tmp.fillna(0) # Somehow, some scaled values become NaN so keep this\n",
    "                \n",
    "                tmp.reset_index(inplace=True)\n",
    "\n",
    "                #tmp1=tmp.sort_values(['diag']).T\n",
    "                tmp=tmp.sort_values(['idh','diag'])\n",
    "                #print(tmp.head())\n",
    "                tmp.drop(['subject'],inplace=True,axis='columns')\n",
    "                tmp.set_index(['sample','idh'],inplace=True)\n",
    "                \n",
    "                diagnoses=tmp.pop('diag')\n",
    "                #lut = dict(zip(diagnoses.unique(), \"rgbcmykb\"))\n",
    "                #col_colors = diagnoses.map(lut)\n",
    "                #print(col_colors)\n",
    "                #print(tmp.columns)\n",
    "                \n",
    "                tmp=tmp.T\n",
    "\n",
    "                #print('pal')\n",
    "                network_pal = sns.cubehelix_palette(len(diagnoses.unique()),\n",
    "                                    light=.9, dark=.1, reverse=True,\n",
    "                                    start=1, rot=-2)\n",
    "                #print(network_pal)\n",
    "                network_pal\n",
    "                #print('lut')\n",
    "                network_lut = dict(zip(diagnoses.unique(), network_pal))\n",
    "                #print(network_lut)\n",
    "                #print('labels')\n",
    "                # Convert the palette to vectors that will be drawn on the side of the matrix\n",
    "                network_labels = diagnoses\n",
    "                #print(network_labels)\n",
    "                network_colors = pd.Series(network_labels, index=tmp.columns).map(network_lut)                \n",
    "                #print('colors')\n",
    "                #print(network_colors)\n",
    "                \n",
    "                # from http://dawnmy.github.io/2016/10/24/Plot-heatmaap-with-side-color-indicating-the-class-of-variables/\n",
    "                # Create a custom palette to identify the networks\n",
    "                #network_pal = sns.cubehelix_palette(len(used_networks),\n",
    "                #                    light=.9, dark=.1, reverse=True,\n",
    "                #                    start=1, rot=-2)\n",
    "                #network_lut = dict(zip(map(str, used_networks), network_pal))\n",
    "\n",
    "                # Convert the palette to vectors that will be drawn on the side of the matrix\n",
    "                #network_labels = df.columns.get_level_values(\"network\")\n",
    "                #network_colors = pd.Series(network_labels, index=df.columns).map(network_lut)                \n",
    "   \n",
    "                # Create a custom colormap for the heatmap values\n",
    "                #cmap = sns.diverging_palette(h_neg=210, h_pos=350, s=90, l=30, as_cmap=True)\n",
    "\n",
    "                # Draw the full plot\n",
    "                #g = sns.clustermap(df.corr(),\n",
    "\n",
    "                #                 # Turn off the clustering\n",
    "                #                 row_cluster=False, col_cluster=False,\n",
    "\n",
    "                #                  # Add colored class labels\n",
    "                #                  row_colors=network_colors, col_colors=network_colors,\n",
    "\n",
    "                #                  # Make the plot look better when many rows/cols\n",
    "                #                  linewidths=0, xticklabels=False, yticklabels=False)\n",
    "\n",
    "                # Draw the legend bar for the classes                 \n",
    "                #for label in network_labels.unique():\n",
    "                #    g.ax_col_dendrogram.bar(0, 0, color=network_lut[label],\n",
    "                #                            label=label, linewidth=0)\n",
    "                #g.ax_col_dendrogram.legend(loc=\"center\", ncol=5)\n",
    "\n",
    "                \n",
    "                g=sns.clustermap(tmp,col_cluster=False,figsize=(15,20),cmap=\"Spectral\",col_colors=network_colors)\n",
    "                g.fig.suptitle(\"Clustermap 1 : \"+trimmer+\" \"+labkit)\n",
    "                plt.setp(g.ax_heatmap.get_xticklabels(), rotation=75, horizontalalignment='right')\n",
    "\n",
    "                # Draw the legend bar for the classes                 \n",
    "                for label in network_labels.unique():\n",
    "                    g.ax_col_dendrogram.bar(0, 0, color=network_lut[label],\n",
    "                                            label=label, linewidth=0)\n",
    "                g.ax_col_dendrogram.legend(loc=\"center\", ncol=4)\n",
    "                \n",
    "                # Adjust the postion of the main colorbar for the heatmap\n",
    "                #g.cax.set_position([.97, .2, .03, .45])\n",
    "                g.cax.set_position([0, .15, .005, .65])\n",
    "                plt.show()\n",
    "\n",
    "                g=sns.clustermap(tmp,figsize=(15,20),cmap=\"Spectral\",col_colors=network_colors)\n",
    "                g.fig.suptitle(\"Clustermap 2 : \"+trimmer+\" \"+labkit)\n",
    "                plt.setp(g.ax_heatmap.get_xticklabels(), rotation=75, horizontalalignment='right')\n",
    "\n",
    "                # Draw the legend bar for the classes                 \n",
    "                for label in network_labels.unique():\n",
    "                    g.ax_col_dendrogram.bar(0, 0, color=network_lut[label],\n",
    "                                            label=label, linewidth=0)\n",
    "                g.ax_col_dendrogram.legend(loc=\"center\", ncol=4)\n",
    "                                \n",
    "                # Adjust the postion of the main colorbar for the heatmap\n",
    "                # [dist from left,dist from bottom?,width?,height?]\n",
    "                g.cax.set_position([0, .15, .005, .65])\n",
    "                plt.show()  \n",
    "    else:\n",
    "        print(\"Not enough data to produce cluster maps\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Normalize by Trimmed Read Count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfn=df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>subject</th>\n",
       "      <th>Lab kit</th>\n",
       "      <th>trimmer</th>\n",
       "      <th>Raw Read Count</th>\n",
       "      <th>Raw Read Length</th>\n",
       "      <th>Trimmed Read Count</th>\n",
       "      <th>Trimmed Ave Read Length</th>\n",
       "      <th>STAR Aligned to Transcriptome</th>\n",
       "      <th>STAR Aligned to Transcriptome %</th>\n",
       "      <th>...</th>\n",
       "      <th>XR_953246.2</th>\n",
       "      <th>XR_953247.2</th>\n",
       "      <th>XR_953250.2</th>\n",
       "      <th>XR_953252.2</th>\n",
       "      <th>XR_953253.2</th>\n",
       "      <th>XR_953316.2</th>\n",
       "      <th>XR_953317.3</th>\n",
       "      <th>XR_953318.3</th>\n",
       "      <th>XR_953324.3</th>\n",
       "      <th>XR_953325.3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1270784</td>\n",
       "      <td>33.8185</td>\n",
       "      <td>111104</td>\n",
       "      <td>8.74</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372973</td>\n",
       "      <td>38.2306</td>\n",
       "      <td>112499</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>bbduk3</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1372997</td>\n",
       "      <td>38.2314</td>\n",
       "      <td>112492</td>\n",
       "      <td>8.19</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt1</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395891</td>\n",
       "      <td>43.5121</td>\n",
       "      <td>112689</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SFHH005aa</td>\n",
       "      <td>3_5</td>\n",
       "      <td>D-plex</td>\n",
       "      <td>cutadapt2</td>\n",
       "      <td>4038879</td>\n",
       "      <td>35.0646</td>\n",
       "      <td>1395893</td>\n",
       "      <td>43.5142</td>\n",
       "      <td>112696</td>\n",
       "      <td>8.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  124196 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      sample subject Lab kit    trimmer  Raw Read Count Raw Read Length  \\\n",
       "0  SFHH005aa     3_5  D-plex     bbduk1         4038879         35.0646   \n",
       "1  SFHH005aa     3_5  D-plex     bbduk2         4038879         35.0646   \n",
       "2  SFHH005aa     3_5  D-plex     bbduk3         4038879         35.0646   \n",
       "3  SFHH005aa     3_5  D-plex  cutadapt1         4038879         35.0646   \n",
       "4  SFHH005aa     3_5  D-plex  cutadapt2         4038879         35.0646   \n",
       "\n",
       "   Trimmed Read Count Trimmed Ave Read Length STAR Aligned to Transcriptome  \\\n",
       "0             1270784                 33.8185                        111104   \n",
       "1             1372973                 38.2306                        112499   \n",
       "2             1372997                 38.2314                        112492   \n",
       "3             1395891                 43.5121                        112689   \n",
       "4             1395893                 43.5142                        112696   \n",
       "\n",
       "  STAR Aligned to Transcriptome %  ... XR_953246.2 XR_953247.2 XR_953250.2  \\\n",
       "0                            8.74  ...           0           0           0   \n",
       "1                            8.19  ...           0           0           0   \n",
       "2                            8.19  ...           0           0           0   \n",
       "3                            8.07  ...           0           0           0   \n",
       "4                            8.07  ...           0           0           0   \n",
       "\n",
       "  XR_953252.2 XR_953253.2 XR_953316.2 XR_953317.3 XR_953318.3 XR_953324.3  \\\n",
       "0           0           0           0           1           0           0   \n",
       "1           0           0           0           1           0           0   \n",
       "2           0           0           0           1           0           0   \n",
       "3           0           0           0           1           0           0   \n",
       "4           0           0           0           1           0           0   \n",
       "\n",
       "  XR_953325.3  \n",
       "0           0  \n",
       "1           0  \n",
       "2           0  \n",
       "3           0  \n",
       "4           0  \n",
       "\n",
       "[5 rows x 124196 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfn.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'07:30:10'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datetime.now().strftime(\"%H:%M:%S\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-59-ace2004cb918>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdfn\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0melements\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1e9\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mdfn\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0melements\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0mdfn\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Trimmed Read Count'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3039\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_frame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3040\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3041\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3042\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3043\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_setitem_array\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3074\u001b[0m                 )[1]\n\u001b[1;32m   3075\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3076\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mslice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3077\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3078\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_frame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value)\u001b[0m\n\u001b[1;32m   1732\u001b[0m                     \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0milocs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1733\u001b[0m                         \u001b[0;31m# setting with a list, re-coerces\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1734\u001b[0;31m                         \u001b[0misetter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1735\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1736\u001b[0m                 elif (\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36misetter\u001b[0;34m(loc, v)\u001b[0m\n\u001b[1;32m   1691\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1692\u001b[0m                 \u001b[0;31m# reset the sliced object if unique\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1693\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iset_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mser\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1694\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1695\u001b[0m             \u001b[0;31m# we need an iterable, with a ndim of at least 1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_iset_item\u001b[0;34m(self, loc, value)\u001b[0m\n\u001b[1;32m   3099\u001b[0m         \u001b[0;31m#  but the behavior is the same as long as we pass broadcast=False\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3100\u001b[0m         \u001b[0mvalue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sanitize_column\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbroadcast\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3101\u001b[0;31m         \u001b[0mNDFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_iset_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3102\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3103\u001b[0m         \u001b[0;31m# check if we are modifying a copy\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_iset_item\u001b[0;34m(self, loc, value)\u001b[0m\n\u001b[1;32m   3569\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3570\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_iset_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3571\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_mgr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3572\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_clear_item_cache\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3573\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36miset\u001b[0;34m(self, loc, value)\u001b[0m\n\u001b[1;32m   1112\u001b[0m                     \u001b[0mremoved_blknos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblkno\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1113\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1114\u001b[0;31m                     \u001b[0mblk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblk_locs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1115\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_blklocs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mblk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmgr_locs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1116\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/pandas/core/internals/blocks.py\u001b[0m in \u001b[0;36mdelete\u001b[0;34m(self, loc)\u001b[0m\n\u001b[1;32m    335\u001b[0m         \u001b[0mDelete\u001b[0m \u001b[0mgiven\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mblock\u001b[0m \u001b[0;32min\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mplace\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    336\u001b[0m         \"\"\"\n\u001b[0;32m--> 337\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    338\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmgr_locs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmgr_locs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdelete\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    339\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<__array_function__ internals>\u001b[0m in \u001b[0;36mdelete\u001b[0;34m(*args, **kwargs)\u001b[0m\n",
      "\u001b[0;32m~/Library/Python/3.6/lib/python/site-packages/numpy/lib/function_base.py\u001b[0m in \u001b[0;36mdelete\u001b[0;34m(arr, obj, axis)\u001b[0m\n\u001b[1;32m   4407\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4408\u001b[0m         \u001b[0mslobj\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkeep\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4409\u001b[0;31m         \u001b[0mnew\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mslobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4410\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4411\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mwrap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "dfn[elements]=(1e9*dfn[elements].values)/dfn[['Trimmed Read Count']].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "datetime.now().strftime(\"%H:%M:%S\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfn.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "datetime.now().strftime(\"%H:%M:%S\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for c in ['CDKN2A','CDKN2A-DT','CDKN2AIP','CDKN2AIPNL','CDKN2B','CDKN2B-AS1',\n",
    "    'EGFR','EGFR-AS1','IDH1','IDH1-AS1','IDH2','IDH2-DT','NF1','NOTCH1',\n",
    "    'RB1','RB1CC1','RB1-DT','TERT','TP53']:\n",
    "        if( c not in dfn.columns ):\n",
    "            print(\"Missing \"+c) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "datetime.now().strftime(\"%H:%M:%S\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Select Genes (if using Genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ttests_boxplots_and_heatmaps(dfn,['CDKN2A','CDKN2A-DT','CDKN2AIP','CDKN2AIPNL','CDKN2B','CDKN2B-AS1','EGFR',\n",
    "                                  'EGFR-AS1','IDH1','IDH1-AS1','IDH2','IDH2-DT','NF1','NOTCH1','RB1','RB1CC1',\n",
    "                                  'RB1-DT','TERT','TP53'],box_p=0.1,heat_p=0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "datetime.now().strftime(\"%H:%M:%S\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Elements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ttests_boxplots_and_heatmaps(dfn,elements,box_p=0.01,heat_p=0.1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "datetime.now().strftime(\"%H:%M:%S\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.decomposition import PCA\n",
    "#alldiags=dfn['diag'].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "for labkit in dfn['Lab kit'].unique():\n",
    "    #\"D-plex\",\"Lexogen\":\n",
    "    for trimmer in 'bbduk2','cutadapt2':\n",
    "        #localdf['Trimmer'].unique():\n",
    "        #pcadf=dfn[(dfn[\"Lab kit\"]==labkit) & (dfn[\"trimmer\"]==trimmer).copy()\n",
    "        pcadf=dfn[(dfn[\"Lab kit\"]==labkit) & (dfn[\"trimmer\"]==trimmer) & dfn['diag'].isin(diags)].copy()\n",
    "\n",
    "        scaler = StandardScaler()\n",
    "        scaler.fit(pcadf[elements])\n",
    "        scaled_data = scaler.transform(pcadf[elements])\n",
    "        #scaled_data[0]\n",
    "        pca = PCA(n_components=2)\n",
    "        pca.fit(scaled_data)\n",
    "        x_pca = pca.transform(scaled_data)\n",
    "        #pcadf['diag_index']=pcadf['diag'].apply(lambda x: alldiags.index(x))\n",
    "        pcadf['diag_index']=pcadf['diag'].apply(lambda x: diags.index(x))\n",
    "        fig = plt.figure(figsize = (18,8))\n",
    "        ax = fig.add_subplot(1,1,1) \n",
    "        ax.set_xlabel('Principal Component 1', fontsize = 15)\n",
    "        ax.set_ylabel('Principal Component 2', fontsize = 15)\n",
    "        ax.set_title('2 component PCA', fontsize = 20)\n",
    "        #targets = alldiags\n",
    "        targets = diags\n",
    "        colors = ['r','g','b','c','m','y','k']\n",
    "        for target, color in zip(targets,colors):\n",
    "            indicesToKeep = pcadf['diag'] == target\n",
    "            ax.scatter(x_pca[indicesToKeep, 0], x_pca[indicesToKeep, 1], c = color, s = 50)\n",
    "        ax.legend(targets)\n",
    "        ax.grid()\n",
    "        plt.title(labkit + \" \" + trimmer)\n",
    "        plt.show()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams[\"figure.figsize\"] = [18.0,8.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', default_max_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows',default_max_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.precision', default_precision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
